<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Proteomics Data Analysis (PDA)</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PDA25EBI</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Topics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_quantification_preprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">Quantification - Differential Abundance Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Exercises
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_tutorialPreprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PDA25EBI">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Proteomics Data Analysis (PDA)</h1>

</div>


<hr />
<p><img src="figures/IntroFig.png" width="1440" /></p>
<div id="course-description" class="section level3">
<h3>Course Description</h3>
<p>Mass spectrometry based proteomic experiments generate ever larger
datasets and, as a consequence, complex data interpretation challenges.
This course focuses on the statistical concepts for peptide
identification, quantification, and differential analysis. Moreover,
more advanced experimental designs and blocking will also be introduced.
The core focus will be on shotgun proteomics data, and quantification
using label-free precursor peptide (MS1) ion intensities. The course
will rely exclusively on free and userfriendly opensource tools in
R/Bioconductor. The course will provide a solid basis for beginners, but
will also bring new perspectives to those already familiar with standard
data interpretation procedures in proteomics.</p>
<p>Students can sharpen their background knowledge on Mass Spectrometry,
Proteomics &amp; Bioinformatics for Proteomics here: <a
href="./techvid.html">Mass Spectrometry and Bioinformatics for
Proteomics</a></p>
</div>
<div id="target-audience" class="section level3">
<h3>Target Audience</h3>
<p>This course is oriented towards biologists and bioinformaticians with
a particular interest in differential analysis for quantitative
proteomics.</p>
<p>According to the target audience of the course we either work with a
graphical user interface (GUI) in a R/shiny App msqrob2gui e.g. </p>
<ul>
<li>Proteomics Data Analysis course at the Gulbenkian institute</li>
<li>Proteomics Bioinformatics course of the EBI</li>
<li>BePA introduction: What can proteomics do for you?</li>
</ul>
<p>or with R/markdowns scripts e.g. </p>
<ul>
<li>Bioinformatics Summer School at UCLouvain</li>
<li>Statistical Genomics Course at Ghent University.</li>
</ul>
</div>
<div id="software-data" class="section level3">
<h3>Software &amp; Data</h3>
<ul>
<li><a href="./software.html">Install msqrob2 software</a></li>
<li><a
href="https://github.com/statOmics/PDA/archive/refs/heads/data.zip">Download
data</a></li>
</ul>
<p>Note, that users who develop R/markdown scripts can access data both
from the web or from disk within their scripts. So they do not need to
download the data first. The msqrob2gui Shiny App only works with data
that is available on disk.</p>
<ul>
<li>More information on our tools can be found in our papers <span
class="citation">(L. J. Goeminne, Gevaert, and Clement 2016)</span>,
<span class="citation">(L. J. E. Goeminne et al. 2020)</span>, <span
class="citation">(Sticker et al. 2020)</span> and <span
class="citation">(Vandenbulcke and Clement 2025)</span>. Please refer to
our work when using our tools.</li>
</ul>
</div>
<div id="instructor" class="section level3">
<h3>Instructor</h3>
<ul>
<li><a href="https://statomics.github.io/pages/about.html">Lieven
Clement</a>, Associate Professor of Statistical Genomics, <a
href="https://statomics.github.io/">statOmics</a>, <a
href="https://www.ugent.be/">Ghent University</a>, Belgium</li>
</ul>
</div>
<div id="issues" class="section level3">
<h3>Issues</h3>
<p>If you encounter problems related to the course material
(e.g. package installation problems, bugs in the code, typos, …), please
consider <a href="https://github.com/statOmics/PDA/issues">posting an
issue on GitHub</a>.</p>
<hr />
</div>
<div id="detailed-program" class="section level3">
<h3>Detailed Program</h3>
<ol style="list-style-type: decimal">
<li>Preprocessing &amp; Analysis of Label Free Quantitative Proteomics
Experiments with Simple Designs</li>
</ol>
<ul>
<li>Lecture: <a
href="./pda_quantification_preprocessing.html">Preprocessing</a>, [<a
href="./pda_quantification_preprocessing.pdf">PDF</a>]</li>
<li>Tutorial: <a
href="./pda_tutorialPreprocessing.html">preprocessing</a></li>
<li>Wrap-up: <a
href="./pda_robustSummarisation_peptideModels.html">Peptide-based
Models</a>[<a
href="./pda_robustSummarisation_peptideModels.pdf">PDF</a>]</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Statistical Inference &amp; Analysis of Experiments with Factorial
Designs</li>
</ol>
<ul>
<li>Lecture: <a href="./pda_quantification_inference.html">Differential
Abundance Analysis</a>, [<a
href="./pda_quantification_inference.pdf">PDF</a>]</li>
<li>Tutorial: <a href="pda_tutorialDesign.html">Design</a></li>
</ul>
<hr />
</div>
<div id="acknowledgements" class="section level3">
<h3>Acknowledgements</h3>
<p>A special thanks to Pedro Fernandes, Coordinator of the GTPB
Bioinformatics Training Programme, Instituto Gulbenkian de Ciencia, who
radically changed our view on teaching and immersed us in the teaching
method that we use in this course.</p>
</div>
<div id="license" class="section level3">
<h3>License</h3>
<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This project is licensed under the <a
href="https://creativecommons.org/licenses/by-nc-sa/4.0">Creative
Commons Attribution-NonCommercial-ShareAlike 4.0 International (CC
BY-NC-SA 4.0)</a></p>
<hr />
</div>
<div id="references" class="section level3 unnumbered">
<h3 class="unnumbered">References</h3>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-goeminne2020" class="csl-entry">
Goeminne, L. J. E., A. Sticker, L. Martens, K. Gevaert, and L. Clement.
2020. <span>“<span
class="nocase"><span>M</span><span>S</span>q<span>R</span>ob
<span>T</span>akes the <span>M</span>issing <span>H</span>urdle:
<span>U</span>niting <span>I</span>ntensity- and
<span>C</span>ount-<span>B</span>ased
<span>P</span>roteomics</span>.”</span> <em>Anal Chem</em> 92 (9):
6278–87.
</div>
<div id="ref-goeminne2016" class="csl-entry">
Goeminne, L. J., K. Gevaert, and L. Clement. 2016. <span>“<span
class="nocase"><span>P</span>eptide-level <span>R</span>obust
<span>R</span>idge <span>R</span>egression <span>I</span>mproves
<span>E</span>stimation, <span>S</span>ensitivity, and
<span>S</span>pecificity in <span>D</span>ata-dependent
<span>Q</span>uantitative <span>L</span>abel-free <span>S</span>hotgun
<span>P</span>roteomics</span>.”</span> <em>Mol Cell Proteomics</em> 15
(2): 657–68.
</div>
<div id="ref-sticker2020" class="csl-entry">
Sticker, A., L. Goeminne, L. Martens, and L. Clement. 2020. <span>“<span
class="nocase"><span>R</span>obust <span>S</span>ummarization and
<span>I</span>nference in <span>P</span>roteome-wide
<span>L</span>abel-free <span>Q</span>uantification</span>.”</span>
<em>Mol Cell Proteomics</em> 19 (7): 1209–19.
</div>
<div id="ref-vandenbulcke2025" class="csl-entry">
Vandenbulcke, C., S. Vanderaa, and L. Clement. 2025. <span>“msqrob2TMT:
Robust Linear Mixed Models for Inferring Differential Signals in Tandem
Mass Tag-Based Proteomics.”</span> <em>Molecular &amp; Cellular
Proteomics</em> 24 (3): e10101–1. <a
href="https://doi.org/10.1016/j.mcpro.2025.00101-X">https://doi.org/10.1016/j.mcpro.2025.00101-X</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
