<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>Statistical Methods for Quantitative MS-based Proteomics: Part II. Differential Abundance Analysis</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>


<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PDA25EBI</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Topics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_quantification_preprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_robustSummarisation_peptideModels.html">Wrap-up Preprocessing - Peptide-Level-Models</a>
    </li>
    <li>
      <a href="pda_quantification_inference.html">Quantification - Differential Abundance Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-laptop"></span>
     
    Exercises
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pda_tutorialPreprocessing.html">Quantification - Preprocessing</a>
    </li>
    <li>
      <a href="pda_tutorialDesign.html">Quantification - Design</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/statOmics/PDA25EBI">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Statistical Methods for Quantitative
MS-based Proteomics: Part II. Differential Abundance Analysis</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date"><a href="https://statomics.github.io">statOmics</a>,
Ghent University</h4>

</div>


<p><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a></p>
<p>This is part of the online course <a
href="https://statomics.github.io/PDA/">Proteomics Data Analysis
(PDA)</a></p>
<iframe width="560" height="315" src="https://www.youtube.com/embed/1mhg3BCuEm8" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<ul>
<li><a
href="https://www.youtube.com/watch?v=1mhg3BCuEm8&amp;list=PLZH1hP8_LbJKqnPSS4hxTkn-tQolSCGjP">Playlist
PDA Preprocessing</a></li>
</ul>
<div id="outline" class="section level1 unnumbered">
<h1 class="unnumbered">Outline</h1>
<ul>
<li>Francisella tularensis Example</li>
<li>Hypothesis testing</li>
<li>Multiple testing</li>
<li>Moderated statistics</li>
<li>Experimental design</li>
</ul>
<p>Note, that the R-code is included for learners who are aiming to
develop R/markdown scripts to automate their quantitative proteomics
data analyses. According to the target audience of the course we either
work with a graphical user interface (GUI) in a R/shiny App msqrob2gui
(e.g. Proteomics Bioinformatics course of the EBI and the Proteomics
Data Analysis course at the Gulbenkian institute) or with R/markdowns
scripts (e.g. Bioinformatics Summer School at UCLouvain or the
Statistical Genomics Course at Ghent University).</p>
<hr />
</div>
<div id="francisella-tularensis-experiment" class="section level1"
number="1">
<h1><span class="header-section-number">1</span> Francisella tularensis
experiment</h1>
<iframe width="560" height="315" src="https://www.youtube.com/embed/TW5rk1y7aOc" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p><img src="figures/francisella.jpg" width="50%" /></p>
<p><img src="figures/tularemia_lesion.jpg" width="350" /></p>
<ul>
<li>Pathogen: causes tularemia</li>
<li>Metabolic adaptation key for intracellular life cycle of pathogenic
microorganisms.</li>
<li>Upon entry into host cells quick phasomal escape and active
multiplication in cytosolic compartment.</li>
<li>Franciscella is auxotroph for several amino acids, including
arginine.</li>
<li>Inactivation of arginine transporter delayed bacterial phagosomal
escape and intracellular multiplication.</li>
<li>Experiment to assess difference in proteome using 3 WT vs 3 ArgP KO
mutants</li>
</ul>
<div id="import-the-data-in-r" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Import the data in
R</h2>
<iframe width="560" height="315" src="https://www.youtube.com/embed/2Ljd95IE_AY" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<details>
<summary>
Click to see code
</summary>
<p>
<ol style="list-style-type: decimal">
<li>Load libraries</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(QFeatures)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(msqrob2)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(data.table)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant
that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>peptidesTable <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA/data/quantification/francisella/peptides.txt&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>int64 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">sapply</span>(peptidesTable,class) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> int64) peptidesTable[[j]] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(peptidesTable[[j]])                      </span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in
columnns that start with Intensity. We can retreive the column names
with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>quantCols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity &quot;</span>, <span class="fu">names</span>(peptidesTable))</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="at">assayData =</span> peptidesTable,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">quantCols =</span>  quantCols,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span></code></pre></div>
<pre><code>## Checking arguments.</code></pre>
<pre><code>## Loading data as a &#39;SummarizedExperiment&#39; object.</code></pre>
<pre><code>## Formatting sample annotations (colData).</code></pre>
<pre><code>## Formatting data as a &#39;QFeatures&#39; object.</code></pre>
<pre><code>## Setting assay rownames.</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">rm</span>(peptidesTable)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8795485 469.8   15504024 828.1         NA 15504024 828.1
## Vcells 34574466 263.8   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8795334 469.8   15504024 828.1         NA 15504024 828.1
## Vcells 34567828 263.8   81347764 620.7      16384 81347764 620.7</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Update data with information on design</li>
</ol>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>genotype <span class="ot">&lt;-</span> pe[[<span class="dv">1</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span> </span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="dv">12</span>,<span class="dv">13</span>) <span class="sc">|&gt;</span></span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="fu">as.factor</span>() <span class="sc">|&gt;</span> </span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="fu">relevel</span>(<span class="st">&quot;WT&quot;</span>)</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" tabindex="-1"></a>pe <span class="sc">|&gt;</span> <span class="fu">colData</span>()</span></code></pre></div>
<pre><code>## DataFrame with 6 rows and 1 column
##                          genotype
##                          &lt;factor&gt;
## Intensity 1WT_20_2h_n3_3       WT
## Intensity 1WT_20_2h_n4_3       WT
## Intensity 1WT_20_2h_n5_3       WT
## Intensity 3D8_20_2h_n3_3       D8
## Intensity 3D8_20_2h_n4_3       D8
## Intensity 3D8_20_2h_n5_3       D8</code></pre>
</p>
</details>
</div>
<div id="preprocessing" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Preprocessing</h2>
<details>
<summary>
Click to see code to log-transfrom the data
</summary>
<p>
<ol style="list-style-type: decimal">
<li>Log transform</li>
</ol>
<ul>
<li>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Filtering</li>
</ol>
<ul>
<li>We remove peptides that could not be mapped to a protein or that map
to multiple proteins (the protein identifier contains multiple
identifiers separated by a <code>;</code>).</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>    pe, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;;&quot;</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<ul>
<li>Remove reverse sequences (decoys) and contaminants. Note that this
is indicated by the column names Reverse and depending on the version of
maxQuant with Potential.contaminants or Contaminants.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<ul>
<li>Drop peptides that were identified in less than three sample. We
tolerate the following proportion of NAs: pNA = (n-3)/n. </li>
</ul>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 5740</code></pre>
<p>We keep 5740 peptides upon filtering.</p>
<ol start="3" style="list-style-type: decimal">
<li>Normalization by median centering</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Summarization. We use the standard sumarisation in
aggregateFeatures, which is a robust summarisation method.</li>
</ol>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>, </span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Filter proteins that contain many missing values</li>
</ol>
<p>We want to have at least 4 observed proteins so that most proteins
have at least 2 observations in each group.<br />
So we tolerate a proportion of (n-4)/n NAs.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="st">&quot;protein&quot;</span>]])</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<p>Plot of preprocessed data</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>pe[[<span class="st">&quot;peptideNorm&quot;</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>  <span class="fu">assay</span>() <span class="sc">|&gt;</span></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>  <span class="fu">gather</span>(sample, intensity) <span class="sc">|&gt;</span> </span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="fu">colData</span>(pe)[sample,<span class="st">&quot;genotype&quot;</span>]) <span class="sc">|&gt;</span></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> intensity,<span class="at">group =</span> sample,<span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Peptide-level&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 4421 rows containing non-finite outside the scale range
## (`stat_density()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>pe[[<span class="st">&quot;protein&quot;</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">assay</span>() <span class="sc">|&gt;</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="fu">gather</span>(sample, intensity) <span class="sc">|&gt;</span> </span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="fu">colData</span>(pe)[sample,<span class="st">&quot;genotype&quot;</span>]) <span class="sc">|&gt;</span></span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> intensity,<span class="at">group =</span> sample,<span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Protein-level&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 149 rows containing non-finite outside the scale range
## (`stat_density()`).</code></pre>
<img src="pda_quantification_inference_files/figure-html/unnamed-chunk-16-2.png" width="672" />
</p>
</details>
</div>
<div id="summarized-data-structure" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Summarized data
structure</h2>
<div id="design" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Design</h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>pe <span class="sc">|&gt;</span> </span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>  <span class="fu">colData</span>() <span class="sc">|&gt;</span> </span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">genotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intensity 1WT_20_2h_n3_3</td>
<td align="left">WT</td>
</tr>
<tr class="even">
<td align="left">Intensity 1WT_20_2h_n4_3</td>
<td align="left">WT</td>
</tr>
<tr class="odd">
<td align="left">Intensity 1WT_20_2h_n5_3</td>
<td align="left">WT</td>
</tr>
<tr class="even">
<td align="left">Intensity 3D8_20_2h_n3_3</td>
<td align="left">D8</td>
</tr>
<tr class="odd">
<td align="left">Intensity 3D8_20_2h_n4_3</td>
<td align="left">D8</td>
</tr>
<tr class="even">
<td align="left">Intensity 3D8_20_2h_n5_3</td>
<td align="left">D8</td>
</tr>
</tbody>
</table>
<ul>
<li>WT vs KO</li>
<li>3 vs 3 repeats</li>
</ul>
</div>
<div id="summarized-intensity-matrix" class="section level3"
number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Summarized
intensity matrix</h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a>pe[[<span class="st">&quot;protein&quot;</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>  <span class="fu">assay</span>() <span class="sc">|&gt;</span> </span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span> </span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">kable</span>()</span></code></pre></div>
<table>
<colgroup>
<col width="7%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Intensity 1WT_20_2h_n3_3</th>
<th align="right">Intensity 1WT_20_2h_n4_3</th>
<th align="right">Intensity 1WT_20_2h_n5_3</th>
<th align="right">Intensity 3D8_20_2h_n3_3</th>
<th align="right">Intensity 3D8_20_2h_n4_3</th>
<th align="right">Intensity 3D8_20_2h_n5_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">WP_003013731</td>
<td align="right">-0.2277864</td>
<td align="right">-0.0333727</td>
<td align="right">0.1810282</td>
<td align="right">-0.1946484</td>
<td align="right">0.1347441</td>
<td align="right">0.0522102</td>
</tr>
<tr class="even">
<td align="left">WP_003013909</td>
<td align="right">-0.7603080</td>
<td align="right">-0.8862012</td>
<td align="right">-0.8565867</td>
<td align="right">-0.4951824</td>
<td align="right">-0.6100506</td>
<td align="right">-0.5490239</td>
</tr>
<tr class="odd">
<td align="left">WP_003014068</td>
<td align="right">0.5743424</td>
<td align="right">0.7821886</td>
<td align="right">1.0336891</td>
<td align="right">0.5098828</td>
<td align="right">0.8536632</td>
<td align="right">0.5446984</td>
</tr>
<tr class="even">
<td align="left">WP_003014122</td>
<td align="right">-0.8382825</td>
<td align="right">-1.1131095</td>
<td align="right">-0.9099676</td>
<td align="right">-1.2104299</td>
<td align="right">-1.2575807</td>
<td align="right">-1.3093650</td>
</tr>
<tr class="odd">
<td align="left">WP_003014123</td>
<td align="right">-0.2803634</td>
<td align="right">-0.4062057</td>
<td align="right">-0.3159891</td>
<td align="right">-0.4216095</td>
<td align="right">-0.3453093</td>
<td align="right">-0.6329528</td>
</tr>
<tr class="even">
<td align="left">WP_003014302</td>
<td align="right">1.5696671</td>
<td align="right">1.7486922</td>
<td align="right">1.5869110</td>
<td align="right">1.6919901</td>
<td align="right">1.8142693</td>
<td align="right">1.5398097</td>
</tr>
</tbody>
</table>
<ul>
<li>1025 proteins</li>
</ul>
</div>
<div id="hypothesis-testing-a-single-protein" class="section level3"
number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Hypothesis testing:
a single protein</h3>
<iframe width="560" height="315" src="https://www.youtube.com/embed/myP6SUlSwsM" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div id="t-test" class="section level4" number="1.3.3.1">
<h4><span class="header-section-number">1.3.3.1</span> T-test</h4>
<iframe width="560" height="315" src="https://www.youtube.com/embed/cVw5kdSRZCE" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p><span class="math display">\[
\log_2 \text{FC} = \bar{y}_{p1}-\bar{y}_{p2}
\]</span></p>
<p><span class="math display">\[
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
\]</span></p>
<p><span class="math display">\[
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
\]</span></p>
<p>If we can assume equal variance in both treatment groups:</p>
<p><span class="math display">\[
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
\]</span></p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>WP_003023392 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>    <span class="at">intensity =</span> <span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]][<span class="st">&quot;WP_003023392&quot;</span>,]) <span class="sc">|&gt;</span> </span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>      <span class="fu">c</span>(), </span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>    <span class="at">genotype =</span> <span class="fu">colData</span>(pe)[,<span class="dv">1</span>]) </span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>WP_003023392 <span class="sc">|&gt;</span> </span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype,<span class="at">y=</span>intensity)) <span class="sc">+</span> </span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Protein WP_003023392&quot;</span>)</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p><span class="math display">\[
t=\frac{\log_2\widehat{\text{FC}}}{\text{se}_{\log_2\widehat{\text{FC}}}}=\frac{-1.4}{0.0541}=-25.8
\]</span></p>
<ul>
<li><p>Is t = -25.8 indicating that there is an effect?</p></li>
<li><p>How likely is it to observe t = -25.8 when there is no effect of
the argP KO on the protein expression?</p></li>
</ul>
</div>
<div id="null-hypothesis-h_0-and-alternative-hypothesis-h_1"
class="section level4" number="1.3.3.2">
<h4><span class="header-section-number">1.3.3.2</span> Null hypothesis
(<span class="math inline">\(H_0\)</span>) and alternative hypothesis
(<span class="math inline">\(H_1\)</span>)</h4>
<ul>
<li><p>With data we can never prove a hypothesis (falsification
principle of Popper)</p></li>
<li><p>With data we can only reject a hypothesis</p></li>
<li><p>In general we start from <em>alternative hypothese</em> <span
class="math inline">\(H_1\)</span>: we want to show an effect of the KO
on a protein</p></li>
</ul>
<center>
<span class="math inline">\(H_1\)</span>: On average the protein
abundance in WT is different from that in KO
</center>
<ul>
<li>But, we will assess this by falsifying the opposite:
<center>
<span class="math inline">\(H_0\)</span>: On average the protein
abundance in WT is equal to that in KO&lt;-
</center></li>
</ul>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">t.test</span>(intensity <span class="sc">~</span> genotype, <span class="at">data =</span> WP_003023392, <span class="at">var.equal=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## 
##  Two Sample t-test
## 
## data:  intensity by genotype
## t = 25.828, df = 4, p-value = 1.335e-05
## alternative hypothesis: true difference in means between group WT and group D8 is not equal to 0
## 95 percent confidence interval:
##  1.246966 1.547353
## sample estimates:
## mean in group WT mean in group D8 
##       -0.2641239       -1.6612832</code></pre>
<ul>
<li><p>How likely is it to observe an equal or more extreme effect than
the one observed in the sample when the null hypothesis is
true?</p></li>
<li><p>When we make assumptions about the distribution of our test
statistic we can quantify this probability: <em>p-value</em>. The
p-value will only be calculated correctly if the underlying assumptions
hold!</p></li>
<li><p>When we repeat the experiment, the probability to observe a fold
change for this gene that is more extreme than a 2.63 fold (<span
class="math inline">\(\log_2 FC=-1.4\)</span>) down or up regulation by
random change (if <span class="math inline">\(H_0\)</span> is true) is
13 out of 1 000 000.<br />
</p></li>
<li><p>If the p-value is below a significance threshold <span
class="math inline">\(\alpha\)</span> we reject the null hypothesis.
<em>We control the probability on a false positive result at the <span
class="math inline">\(\alpha\)</span>-level (type I error)</em></p></li>
<li><p>Note, that the p-values are uniform under the null hypothesis,
i.e. when <span class="math inline">\(H_0\)</span> is true all p-values
are equally likely.</p></li>
</ul>
</div>
</div>
</div>
<div id="multiple-hypothesis-testing" class="section level2"
number="1.4">
<h2><span class="header-section-number">1.4</span> Multiple hypothesis
testing</h2>
<iframe width="560" height="315" src="https://www.youtube.com/embed/cLn-CFyA6ps" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<ul>
<li><p>Consider testing DA for all <span
class="math inline">\(m=1066\)</span> proteins simultaneously</p></li>
<li><p>What if we assess each individual test at level <span
class="math inline">\(\alpha\)</span>? <span
class="math inline">\(\rightarrow\)</span> Probability to have a false
positive (FP) among all m simultatenous test <span
class="math inline">\(&gt;&gt;&gt;  \alpha= 0.05\)</span></p></li>
<li><p>Indeed for each non DA protein we have a probability of 5% to
return a FP.</p></li>
<li><p>In a typical experiment the majority of the proteins are non
DA.</p></li>
<li><p>So an upperbound of the expected FP is <span
class="math inline">\(m \times \alpha\)</span> or <span
class="math inline">\(1066 \times 0.05=53\)</span>.</p></li>
</ul>
<p><span class="math inline">\(\rightarrow\)</span> Hence, we are bound
to call many false positive proteins each time we run the
experiment.</p>
<div id="multiple-testing" class="section level3" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Multiple
testing</h3>
<div id="family-wise-error-rate" class="section level4"
number="1.4.1.1">
<h4><span class="header-section-number">1.4.1.1</span> Family-wise error
rate</h4>
<iframe width="560" height="315" src="https://www.youtube.com/embed/IL_eUSyRDRA" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p>The family-wise error rate (FWER) addresses the multiple testing
issue by no longer controlling the individual type I error for each
protein, instead it controls:</p>
<p><span class="math display">\[
   \text{FWER} =
   \text{P}\left[FP \geq 1 \right].
\]</span></p>
<p>The Bonferroni method is widely used to control the type I error:</p>
<ul>
<li>assess each test at <span
class="math display">\[\alpha_\text{adj}=\frac{\alpha}{m}\]</span></li>
<li>or use adjusted p-values and compare them to <span
class="math inline">\(\alpha\)</span>: <span
class="math display">\[p_\text{adj}=\text{min}\left(p \times
m,1\right)\]</span></li>
</ul>
<p>Problem, the method is very conservative!</p>
</div>
<div id="false-discovery-rate" class="section level4" number="1.4.1.2">
<h4><span class="header-section-number">1.4.1.2</span> False discovery
rate</h4>
<iframe width="560" height="315" src="https://www.youtube.com/embed/envDqvEwRcc" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<ul>
<li>FDR: Expected proportion of false positives on the total number of
positives you return.</li>
<li>An FDR of 1% means that on average we expect 1% false positive
proteins in the list of proteins that are called significant.</li>
<li>Defined by Benjamini and Hochberg in their seminal paper Benjamini,
Y. and Hochberg, Y. (1995). “Controlling the false discovery rate: a
practical and powerful approach to multiple testing”. Journal of the
Royal Statistical Society Series B, 57 (1): 289–300.</li>
</ul>
<p>The <strong>False Discovery Proportion (FDP)</strong> is the fraction
of false positives that are returned, i.e. </p>
<p><span class="math display">\[
FDP = \frac{FP}{R}
\]</span></p>
<ul>
<li>However, this quantity cannot be observed because in practice we
only know the number of proteins for which we rejected <span
class="math inline">\(H_0\)</span>, <span
class="math inline">\(R\)</span>.</li>
<li>But, we do not know the number of false positives, <span
class="math inline">\(FP\)</span>.</li>
</ul>
<p>Therefore, Benjamini and Hochberg, 1995, defined The <strong>False
Discovery Rate (FDR)</strong> as <span class="math display">\[
   \text{FDR} = \text{E}\left[\frac{FP}{R}\right]
=\text{E}\left[\text{FDP}\right]
\]</span> the expected FDP.</p>
<ul>
<li>Controlling the FDR allows for more discoveries (i.e. longer lists
with significant results), while the fraction of false discoveries among
the significant results in well controlled on average. As a consequence,
more of the true positive hypotheses will be detected.</li>
</ul>
</div>
<div id="intuition-of-bh-fdr-procedure" class="section level4"
number="1.4.1.3">
<h4><span class="header-section-number">1.4.1.3</span> Intuition of
BH-FDR procedure</h4>
<p>Consider <span class="math inline">\(m = 1000\)</span> tests</p>
<ul>
<li><p>Suppose that a researcher rejects all null hypotheses for which
<span class="math inline">\(p &lt; 0.01\)</span>.</p></li>
<li><p>If we use <span class="math inline">\(p &lt; 0.01\)</span>, we
expect <span class="math inline">\(0.01 \times m_0\)</span> tests to
return false positives.</p></li>
<li><p>A conservative estimate of the number of false positives that we
can expect can be obtained by considering that the null hypotheses are
true for all features, <span class="math inline">\(m_0 = m
=  1000\)</span>.</p></li>
<li><p>We then would expect <span class="math inline">\(0.01 \times 1000
= 10\)</span> false positives (<span
class="math inline">\(FP=10\)</span>).</p></li>
<li><p>Suppose that the researcher found 200 genes with <span
class="math inline">\(p&lt;0.01\)</span> (<span
class="math inline">\(R=200\)</span>).</p></li>
<li><p>The proportion of false positive results (FDP = false positive
proportion) among the list of <span class="math inline">\(R=200\)</span>
genes can then be estimated as <span class="math display">\[
\widehat{\text{FDP}}=\frac{FP}{R}=\frac{10}{200}=\frac{0.01 \times
1000}{200} = 0.05.
\]</span></p></li>
</ul>
</div>
<div
id="benjamini-and-hochberg-1995-procedure-for-controlling-the-fdr-at-alpha"
class="section level4" number="1.4.1.4">
<h4><span class="header-section-number">1.4.1.4</span> Benjamini and
Hochberg (1995) procedure for controlling the FDR at <span
class="math inline">\(\alpha\)</span></h4>
<ol style="list-style-type: decimal">
<li><p>Let <span class="math inline">\(p_{(1)}\leq \ldots \leq
p_{(m)}\)</span> denote the ordered <span
class="math inline">\(p\)</span>-values.</p></li>
<li><p>Find the largest integer <span class="math inline">\(k\)</span>
so that <span class="math display">\[
\frac{p_{(k)} \times m}{k} \leq \alpha
\]</span> <span class="math display">\[\text{or}\]</span> <span
class="math display">\[
p_{(k)} \leq k \times \alpha/m
\]</span></p></li>
<li><p>If such a <span class="math inline">\(k\)</span> exists, reject
the <span class="math inline">\(k\)</span> null hypotheses associated
with <span class="math inline">\(p_{(1)}, \ldots, p_{(k)}\)</span>.
Otherwise none of the null hypotheses is rejected.</p></li>
</ol>
<p>The adjusted <span class="math inline">\(p\)</span>-value (also known
as the <span class="math inline">\(q\)</span>-value in FDR literature):
<span class="math display">\[
   q_{(i)}=\tilde{p}_{(i)} = \min\left[\min_{j=i,\ldots, m}\left(m
p_{(j)}/j\right), 1 \right].
\]</span> In the hypothetical example above: <span
class="math inline">\(k=200\)</span>, <span
class="math inline">\(p_{(k)}=0.01\)</span>, <span
class="math inline">\(m=1000\)</span> and <span
class="math inline">\(\alpha=0.05\)</span>.</p>
</div>
<div id="francisella-example" class="section level4" number="1.4.1.5">
<h4><span class="header-section-number">1.4.1.5</span> Francisella
Example</h4>
<iframe width="560" height="315" src="https://www.youtube.com/embed/B3BmnOLkYg4" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a>ttestMx <span class="ot">&lt;-</span> <span class="cf">function</span>(y,group) {</span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a>    test <span class="ot">&lt;-</span> <span class="fu">try</span>(<span class="fu">t.test</span>(y[group],y[<span class="sc">!</span>group],<span class="at">var.equal=</span><span class="cn">TRUE</span>),<span class="at">silent=</span><span class="cn">TRUE</span>)</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is</span>(test,<span class="st">&quot;try-error&quot;</span>)) {</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">c</span>(<span class="at">log2FC=</span><span class="cn">NA</span>,<span class="at">se=</span><span class="cn">NA</span>,<span class="at">tstat=</span><span class="cn">NA</span>,<span class="at">p=</span><span class="cn">NA</span>))</span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>      <span class="fu">return</span>(<span class="fu">c</span>(<span class="at">log2FC=</span> (test<span class="sc">$</span>estimate<span class="sc">%*%</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="sc">-</span><span class="dv">1</span>)),<span class="at">se=</span>test<span class="sc">$</span>stderr,<span class="at">tstat=</span>test<span class="sc">$</span>statistic,<span class="at">pval=</span>test<span class="sc">$</span>p.value))</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>      }</span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a> }</span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a> </span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a> res <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb40-11"><a href="#cb40-11" tabindex="-1"></a>    <span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]]), </span>
<span id="cb40-12"><a href="#cb40-12" tabindex="-1"></a>    <span class="dv">1</span>, </span>
<span id="cb40-13"><a href="#cb40-13" tabindex="-1"></a>    ttestMx,</span>
<span id="cb40-14"><a href="#cb40-14" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">colData</span>(pe)<span class="sc">$</span>genotype<span class="sc">==</span><span class="st">&quot;D8&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb40-15"><a href="#cb40-15" tabindex="-1"></a>  <span class="fu">t</span>() </span>
<span id="cb40-16"><a href="#cb40-16" tabindex="-1"></a> </span>
<span id="cb40-17"><a href="#cb40-17" tabindex="-1"></a> <span class="fu">colnames</span>(res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;logFC&quot;</span>,<span class="st">&quot;se&quot;</span>,<span class="st">&quot;tstat&quot;</span>,<span class="st">&quot;pval&quot;</span>)</span>
<span id="cb40-18"><a href="#cb40-18" tabindex="-1"></a> </span>
<span id="cb40-19"><a href="#cb40-19" tabindex="-1"></a> res <span class="ot">&lt;-</span> res <span class="sc">|&gt;</span> </span>
<span id="cb40-20"><a href="#cb40-20" tabindex="-1"></a>   <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb40-21"><a href="#cb40-21" tabindex="-1"></a>   <span class="fu">na.exclude</span>() <span class="sc">|&gt;</span> </span>
<span id="cb40-22"><a href="#cb40-22" tabindex="-1"></a>   <span class="fu">arrange</span>(pval)</span>
<span id="cb40-23"><a href="#cb40-23" tabindex="-1"></a> </span>
<span id="cb40-24"><a href="#cb40-24" tabindex="-1"></a> res<span class="sc">$</span>adjPval <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res<span class="sc">$</span>pval, <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb40-25"><a href="#cb40-25" tabindex="-1"></a> </span>
<span id="cb40-26"><a href="#cb40-26" tabindex="-1"></a> alpha <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb40-27"><a href="#cb40-27" tabindex="-1"></a></span>
<span id="cb40-28"><a href="#cb40-28" tabindex="-1"></a> res<span class="sc">$</span>adjAlphaForm <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(res),<span class="st">&quot; x &quot;</span>,alpha,<span class="st">&quot;/&quot;</span>,<span class="fu">nrow</span>(res))</span>
<span id="cb40-29"><a href="#cb40-29" tabindex="-1"></a></span>
<span id="cb40-30"><a href="#cb40-30" tabindex="-1"></a> res<span class="sc">$</span>adjAlpha <span class="ot">&lt;-</span> alpha <span class="sc">*</span> (<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(res))<span class="sc">/</span><span class="fu">nrow</span>(res) </span>
<span id="cb40-31"><a href="#cb40-31" tabindex="-1"></a></span>
<span id="cb40-32"><a href="#cb40-32" tabindex="-1"></a> res<span class="sc">$</span><span class="st">&quot;pval &lt; adjAlpha&quot;</span> <span class="ot">&lt;-</span> res<span class="sc">$</span>pval <span class="sc">&lt;</span> res<span class="sc">$</span>adjAlpha </span>
<span id="cb40-33"><a href="#cb40-33" tabindex="-1"></a></span>
<span id="cb40-34"><a href="#cb40-34" tabindex="-1"></a> res<span class="sc">$</span><span class="st">&quot;adjPval &lt; alpha&quot;</span> <span class="ot">&lt;-</span> res<span class="sc">$</span>adjPval <span class="sc">&lt;</span> alpha </span></code></pre></div>
</p>
</details>
<p>FWER: Bonferroni method:<span class="math inline">\(\alpha_\text{adj}
= \alpha/m = 0.05 / 1025= 5\times 10^{-5}\)</span></p>
<table>
<colgroup>
<col width="13%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
<col width="14%" />
<col width="10%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">pval</th>
<th align="right">adjPval</th>
<th align="left">adjAlphaForm</th>
<th align="right">adjAlpha</th>
<th align="left">pval &lt; adjAlpha</th>
<th align="left">adjPval &lt; alpha</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">WP_003023392</td>
<td align="right">-1.3971592</td>
<td align="right">0.0000134</td>
<td align="right">0.0136840</td>
<td align="left">1 x 0.05/1025</td>
<td align="right">0.0000488</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">WP_011733588</td>
<td align="right">-0.4018518</td>
<td align="right">0.0000418</td>
<td align="right">0.0214166</td>
<td align="left">2 x 0.05/1025</td>
<td align="right">0.0000976</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">WP_003038430</td>
<td align="right">-0.4024957</td>
<td align="right">0.0001814</td>
<td align="right">0.0408656</td>
<td align="left">3 x 0.05/1025</td>
<td align="right">0.0001463</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">WP_003026016</td>
<td align="right">-1.0493276</td>
<td align="right">0.0001834</td>
<td align="right">0.0408656</td>
<td align="left">4 x 0.05/1025</td>
<td align="right">0.0001951</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">WP_003014552</td>
<td align="right">-0.9536835</td>
<td align="right">0.0001993</td>
<td align="right">0.0408656</td>
<td align="left">5 x 0.05/1025</td>
<td align="right">0.0002439</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">WP_011733645</td>
<td align="right">0.3838435</td>
<td align="right">0.0002817</td>
<td align="right">0.0481190</td>
<td align="left">6 x 0.05/1025</td>
<td align="right">0.0002927</td>
<td align="left">TRUE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">WP_003040849</td>
<td align="right">-1.2473865</td>
<td align="right">0.0004301</td>
<td align="right">0.0629754</td>
<td align="left">7 x 0.05/1025</td>
<td align="right">0.0003415</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">WP_003038940</td>
<td align="right">-0.2386737</td>
<td align="right">0.0005609</td>
<td align="right">0.0718619</td>
<td align="left">8 x 0.05/1025</td>
<td align="right">0.0003902</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">…</td>
<td align="right">…</td>
<td align="right">…</td>
<td align="right">…</td>
<td align="left">…</td>
<td align="right">…</td>
<td align="left">…</td>
<td align="left">…</td>
</tr>
<tr class="even">
<td align="left">WP_003040562</td>
<td align="right">0.0039480</td>
<td align="right">0.9976429</td>
<td align="right">0.9985797</td>
<td align="left">1065 x 0.05/1066</td>
<td align="right">0.0499531</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">WP_003041130</td>
<td align="right">0.0002941</td>
<td align="right">0.9992812</td>
<td align="right">0.9992812</td>
<td align="left">1066 x 0.05/1066</td>
<td align="right">0.05</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
</div>
<div id="results" class="section level4" number="1.4.1.6">
<h4><span class="header-section-number">1.4.1.6</span> Results</h4>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a>volcanoT <span class="ot">&lt;-</span> res <span class="sc">|&gt;</span> </span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() </span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a>volcanoT</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="moderated-statistics" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Moderated
Statistics</h2>
<iframe width="560" height="315" src="https://www.youtube.com/embed/_Q11LXDy0xU" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p>Problems with ordinary t-test</p>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a>problemPlots <span class="ot">&lt;-</span> <span class="fu">list</span>() </span>
<span id="cb43-2"><a href="#cb43-2" tabindex="-1"></a>problemPlots[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> res <span class="sc">|&gt;</span> </span>
<span id="cb43-3"><a href="#cb43-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> se, <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb43-4"><a href="#cb43-4" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() </span>
<span id="cb43-7"><a href="#cb43-7" tabindex="-1"></a></span>
<span id="cb43-8"><a href="#cb43-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb43-9"><a href="#cb43-9" tabindex="-1"></a>{</span>
<span id="cb43-10"><a href="#cb43-10" tabindex="-1"></a>problemPlots[[i]] <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe) <span class="sc">|&gt;</span> </span>
<span id="cb43-11"><a href="#cb43-11" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb43-12"><a href="#cb43-12" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">intensity =</span> pe[[<span class="st">&quot;protein&quot;</span>]][<span class="fu">rownames</span>(res)[i],] <span class="sc">|&gt;</span> </span>
<span id="cb43-13"><a href="#cb43-13" tabindex="-1"></a>             <span class="fu">assay</span>() <span class="sc">|&gt;</span> </span>
<span id="cb43-14"><a href="#cb43-14" tabindex="-1"></a>             <span class="fu">c</span>()</span>
<span id="cb43-15"><a href="#cb43-15" tabindex="-1"></a>           ) <span class="sc">|&gt;</span> </span>
<span id="cb43-16"><a href="#cb43-16" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>genotype,<span class="at">y=</span>intensity)) <span class="sc">+</span></span>
<span id="cb43-17"><a href="#cb43-17" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb43-18"><a href="#cb43-18" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb43-19"><a href="#cb43-19" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">rownames</span>(res)[i])</span>
<span id="cb43-20"><a href="#cb43-20" tabindex="-1"></a>}</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a>problemPlots</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<pre><code>## 
## [[2]]</code></pre>
<pre><code>## Warning: Removed 6 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-28-2.png" width="672" /></p>
<pre><code>## 
## [[3]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-28-3.png" width="672" /></p>
<p>A general class of moderated test statistics is given by <span
class="math display">\[
   T_g^{mod} = \frac{\bar{Y}_{g1} - \bar{Y}_{g2}}{C \quad \tilde{S}_g} ,
\]</span> where <span class="math inline">\(\tilde{S}_g\)</span> is a
moderated standard deviation estimate.</p>
<ul>
<li><span class="math inline">\(C\)</span> is a constant depending on
the design e.g. <span
class="math inline">\(\sqrt{1/{n_1}+1/n_2}\)</span> for a t-test and of
another form for linear models.</li>
<li><span class="math inline">\(\tilde{S}_g=S_g+S_0\)</span>: add small
positive constant to denominator of t-statistic.</li>
<li>This can be adopted in Perseus.</li>
</ul>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a>simI<span class="ot">&lt;-</span><span class="fu">sapply</span>(res<span class="sc">$</span>se<span class="sc">/</span><span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span><span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>),<span class="cf">function</span>(n,mean,sd) <span class="fu">rnorm</span>(n,mean,sd),<span class="at">n=</span><span class="dv">6</span>,<span class="at">mean=</span><span class="dv">0</span>) <span class="sc">|&gt;</span> </span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a>resSim <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>    simI, </span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>    <span class="dv">1</span>, </span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a>    ttestMx,</span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">colData</span>(pe)<span class="sc">$</span>genotype<span class="sc">==</span><span class="st">&quot;D8&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>  <span class="fu">t</span>() </span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a> <span class="fu">colnames</span>(resSim) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;logFC&quot;</span>,<span class="st">&quot;se&quot;</span>,<span class="st">&quot;tstat&quot;</span>,<span class="st">&quot;pval&quot;</span>)</span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a> resSim <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(resSim)</span>
<span id="cb49-12"><a href="#cb49-12" tabindex="-1"></a> tstatSimPlot <span class="ot">&lt;-</span> resSim <span class="sc">|&gt;</span> </span>
<span id="cb49-13"><a href="#cb49-13" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>tstat)) <span class="sc">+</span></span>
<span id="cb49-14"><a href="#cb49-14" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y=</span>..density.., <span class="at">fill=</span>..count..),<span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb49-15"><a href="#cb49-15" tabindex="-1"></a>     <span class="fu">stat_function</span>(<span class="at">fun=</span>dt,</span>
<span id="cb49-16"><a href="#cb49-16" tabindex="-1"></a>    <span class="at">color=</span><span class="st">&quot;red&quot;</span>,</span>
<span id="cb49-17"><a href="#cb49-17" tabindex="-1"></a>    <span class="at">args=</span><span class="fu">list</span>(<span class="at">df=</span><span class="dv">4</span>)) <span class="sc">+</span> </span>
<span id="cb49-18"><a href="#cb49-18" tabindex="-1"></a>   <span class="fu">ylim</span>(<span class="dv">0</span>,.<span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb49-19"><a href="#cb49-19" tabindex="-1"></a>   <span class="fu">ggtitle</span>(<span class="st">&quot;t-statistic&quot;</span>)</span>
<span id="cb49-20"><a href="#cb49-20" tabindex="-1"></a></span>
<span id="cb49-21"><a href="#cb49-21" tabindex="-1"></a> </span>
<span id="cb49-22"><a href="#cb49-22" tabindex="-1"></a> resSim<span class="sc">$</span>C <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span><span class="sc">+</span><span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>) </span>
<span id="cb49-23"><a href="#cb49-23" tabindex="-1"></a> resSim<span class="sc">$</span>sd <span class="ot">&lt;-</span> resSim<span class="sc">$</span>se<span class="sc">/</span>resSim<span class="sc">$</span>C </span>
<span id="cb49-24"><a href="#cb49-24" tabindex="-1"></a> tstatSimPerseus <span class="ot">&lt;-</span> resSim <span class="sc">|&gt;</span> </span>
<span id="cb49-25"><a href="#cb49-25" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>logFC<span class="sc">/</span>((sd<span class="fl">+.1</span>)<span class="sc">*</span>C))) <span class="sc">+</span></span>
<span id="cb49-26"><a href="#cb49-26" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y=</span>..density.., <span class="at">fill=</span>..count..),<span class="at">bins=</span><span class="dv">30</span>) <span class="sc">+</span></span>
<span id="cb49-27"><a href="#cb49-27" tabindex="-1"></a>     <span class="fu">stat_function</span>(<span class="at">fun=</span>dt,</span>
<span id="cb49-28"><a href="#cb49-28" tabindex="-1"></a>                   <span class="at">color=</span><span class="st">&quot;red&quot;</span>,</span>
<span id="cb49-29"><a href="#cb49-29" tabindex="-1"></a>                  <span class="at">args=</span><span class="fu">list</span>(<span class="at">df=</span><span class="dv">4</span>)) <span class="sc">+</span> </span>
<span id="cb49-30"><a href="#cb49-30" tabindex="-1"></a>     <span class="fu">ylim</span>(<span class="dv">0</span>,.<span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb49-31"><a href="#cb49-31" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Perseus&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(tstatSimPlot,tstatSimPerseus,<span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 1 row containing missing values or values outside the scale range
## (`geom_bar()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<ul>
<li>The choice of <span class="math inline">\(S_0\)</span> in Perseus is
ad hoc and the t-statistic is no-longer t-distributed.</li>
<li>Permutation test, but is difficult for more complex designs.</li>
<li>Allows for Data Dredging because user can choose <span
class="math inline">\(S_0\)</span></li>
</ul>
<div id="empirical-bayes" class="section level3" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> Empirical
Bayes</h3>
<p><img src="figures/limmaShrinkage.png" width="50%" /></p>
<p>Figure courtesy to Rafael Irizarry</p>
<p><span class="math display">\[
   T_g^{mod} = \frac{\bar{Y}_{g1} - \bar{Y}_{g2}}{C \quad \tilde{S}_g} ,
\]</span></p>
<ul>
<li><strong>empirical Bayes</strong> theory provides formal framework
for borrowing strength across proteins,</li>
<li>Implemented in popular bioconductor package <strong>limma</strong>
and <strong>msqrob2</strong></li>
</ul>
<p><span class="math display">\[
  \tilde{S}_g=\sqrt{\frac{d_gS_g^2+d_0S_0^2}{d_g+d_0}},
\]</span></p>
<ul>
<li><span class="math inline">\(S_0^2\)</span>: common variance (over
all proteins)</li>
<li>Moderated t-statistic is t-distributed with <span
class="math inline">\(d_0+d_g\)</span> degrees of freedom.</li>
<li>Note that the degrees of freedom increase by borrowing strength
across proteins!</li>
</ul>
<details>
<summary>
Click to see the code
</summary>
<p>
<ol style="list-style-type: decimal">
<li>We model the protein level expression values using the
<code>msqrob</code> function. By default <code>msqrob2</code> estimates
the model parameters using robust regression.</li>
</ol>
<p>We will model the data with a different group mean for every
genotype. The group is incoded in the variable <code>genotype</code> of
the colData. We can specify this model by using a formula with the
factor <code>genotype</code> as its predictor:
<code>formula = ~genotype</code>.</p>
<p>Note, that a formula always starts with a symbol ‘~’.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span>genotype)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Inference</li>
</ol>
<p>We first explore the design of the model that we specified using the
the package <code>ExploreModelMatrix</code></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb53-2"><a href="#cb53-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span>genotype)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<p>We have two model parameters, the (Intercept) and genotypeD8. This
results in a model with two group means:</p>
<ol style="list-style-type: decimal">
<li>For the wild type (WT) the expected value (mean) of the log2
transformed intensity y for a protein will be modelled using</li>
</ol>
<p><span class="math display">\[\text{E}[Y\vert
\text{genotype}=\text{WT}] = \text{(Intercept)}\]</span></p>
<ol start="2" style="list-style-type: decimal">
<li>For the knockout genotype D8 the expected value (mean) of the log2
transformed intensity y for a protein will be modelled using</li>
</ol>
<p><span class="math display">\[\text{E}[Y\vert
\text{genotype}=\text{D8}] = \text{(Intercept)} +
\text{genotypeD8}\]</span></p>
<p>The average log2FC between D8 and WT is thus <span
class="math display">\[\log_2\text{FC}_{D8-WT}= \text{E}[Y\vert
\text{genotype}=\text{D8}] - \text{E}[Y\vert \text{genotype}=\text{WT}]
= \text{genotypeD8}
\]</span></p>
<p>Hence, assessing the null hypothesis that there is no differential
abundance between D8 and WT can be reformulated as</p>
<p><span class="math display">\[H_0:  \text{genotypeD8}=0\]</span> We
can implement a hypothesis test for each protein in msqrob2 using the
code below:</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;genotypeD8 = 0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;genotypeD8&quot;</span>))</span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
<p>We can show the list with all significant DE proteins at the 5% FDR
using</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>genotypeD8 <span class="sc">|&gt;</span> </span>
<span id="cb56-2"><a href="#cb56-2" tabindex="-1"></a>  <span class="fu">arrange</span>(pval) <span class="sc">|&gt;</span></span>
<span id="cb56-3"><a href="#cb56-3" tabindex="-1"></a>  <span class="fu">filter</span>(adjPval<span class="sc">&lt;</span><span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>##                   logFC         se       df          t         pval     adjPval
## WP_003023392 -1.3971592 0.09349966 6.397042 -14.942933 3.254588e-06 0.003244824
## WP_003040849 -1.2473865 0.12404946 6.397042 -10.055557 3.736229e-05 0.015936958
## WP_003014552 -0.9536835 0.10126476 6.397042  -9.417724 5.550408e-05 0.015936958
## WP_003026016 -1.0283106 0.10528137 6.034774  -9.767261 6.393965e-05 0.015936958
## WP_003033719 -1.1740374 0.13543520 6.186857  -8.668628 1.098307e-04 0.021900233</code></pre>
<p>We can also visualise the results using a volcanoplot</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" tabindex="-1"></a>volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb58-2"><a href="#cb58-2" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>genotypeD8,</span>
<span id="cb58-3"><a href="#cb58-3" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb58-4"><a href="#cb58-4" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb58-5"><a href="#cb58-5" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb58-6"><a href="#cb58-6" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb58-7"><a href="#cb58-7" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb58-8"><a href="#cb58-8" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;msqrob2&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(</span>
<span id="cb59-2"><a href="#cb59-2" tabindex="-1"></a>  volcanoT <span class="sc">+</span>    </span>
<span id="cb59-3"><a href="#cb59-3" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb59-4"><a href="#cb59-4" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;ordinary t-test&quot;</span>),</span>
<span id="cb59-5"><a href="#cb59-5" tabindex="-1"></a>  volcano <span class="sc">+</span>     </span>
<span id="cb59-6"><a href="#cb59-6" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">3</span>,<span class="dv">3</span>)</span>
<span id="cb59-7"><a href="#cb59-7" tabindex="-1"></a>,<span class="at">nrow=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 28 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<ul>
<li><p>The volcano plot opens up when using the EB variance
estimator</p></li>
<li><p>Borrowing strength to estimate the variance using empirical Bayes
solves the issue of returning proteins with a low fold change as
significant due to a low variance.</p></li>
</ul>
</div>
<div id="shrinkage-of-the-variance-and-moderated-t-statistics"
class="section level3" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Shrinkage of the
variance and moderated t-statistics</h3>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" tabindex="-1"></a><span class="fu">qplot</span>(</span>
<span id="cb61-2"><a href="#cb61-2" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels,getSigma),</span>
<span id="cb61-3"><a href="#cb61-3" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels,getSigmaPosterior)) <span class="sc">+</span></span>
<span id="cb61-4"><a href="#cb61-4" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;SD&quot;</span>) <span class="sc">+</span></span>
<span id="cb61-5"><a href="#cb61-5" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;moderated SD&quot;</span>) <span class="sc">+</span></span>
<span id="cb61-6"><a href="#cb61-6" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>,<span class="at">slope =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb61-7"><a href="#cb61-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> ) </span></code></pre></div>
<pre><code>## Warning: Removed 28 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
<ul>
<li>Small variances are shrunken towards the common variance resulting
in large EB variance estimates</li>
<li>Large variances are shrunken towards the common variance resulting
in smaller EB variance estimates</li>
<li>Pooled degrees of freedom of the EB variance estimator are larger
because information is borrowed across proteins to estimate the
variance</li>
</ul>
</div>
</div>
<div id="plots" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Plots</h2>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>genotypeD8 <span class="sc">|&gt;</span></span>
<span id="cb63-2"><a href="#cb63-2" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;protein&quot;</span>) <span class="sc">|&gt;</span></span>
<span id="cb63-3"><a href="#cb63-3" tabindex="-1"></a>    <span class="fu">filter</span>(adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb63-4"><a href="#cb63-4" tabindex="-1"></a>    <span class="fu">pull</span>(protein)</span>
<span id="cb63-5"><a href="#cb63-5" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[sigNames, ])</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" tabindex="-1"></a><span class="cf">for</span> (protName <span class="cf">in</span> sigNames)</span>
<span id="cb64-2"><a href="#cb64-2" tabindex="-1"></a>    {</span>
<span id="cb64-3"><a href="#cb64-3" tabindex="-1"></a>        pePlot <span class="ot">&lt;-</span> pe[protName, , <span class="fu">c</span>(<span class="st">&quot;peptideNorm&quot;</span>, <span class="st">&quot;protein&quot;</span>)]</span>
<span id="cb64-4"><a href="#cb64-4" tabindex="-1"></a>        pePlotDf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">longForm</span>(pePlot))</span>
<span id="cb64-5"><a href="#cb64-5" tabindex="-1"></a>        pePlotDf<span class="sc">$</span>assay <span class="ot">&lt;-</span> <span class="fu">factor</span>(pePlotDf<span class="sc">$</span>assay,</span>
<span id="cb64-6"><a href="#cb64-6" tabindex="-1"></a>            <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;peptideNorm&quot;</span>, <span class="st">&quot;protein&quot;</span>)</span>
<span id="cb64-7"><a href="#cb64-7" tabindex="-1"></a>        )</span>
<span id="cb64-8"><a href="#cb64-8" tabindex="-1"></a>        pePlotDf<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(<span class="fu">colData</span>(pePlot)[pePlotDf<span class="sc">$</span>colname, <span class="st">&quot;genotype&quot;</span>])</span>
<span id="cb64-9"><a href="#cb64-9" tabindex="-1"></a></span>
<span id="cb64-10"><a href="#cb64-10" tabindex="-1"></a>        <span class="co"># plotting</span></span>
<span id="cb64-11"><a href="#cb64-11" tabindex="-1"></a>        p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb64-12"><a href="#cb64-12" tabindex="-1"></a>            <span class="at">data =</span> pePlotDf,</span>
<span id="cb64-13"><a href="#cb64-13" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> colname, <span class="at">y =</span> value, <span class="at">group =</span> rowname)</span>
<span id="cb64-14"><a href="#cb64-14" tabindex="-1"></a>        ) <span class="sc">+</span></span>
<span id="cb64-15"><a href="#cb64-15" tabindex="-1"></a>            <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb64-16"><a href="#cb64-16" tabindex="-1"></a>            <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb64-17"><a href="#cb64-17" tabindex="-1"></a>            <span class="fu">facet_grid</span>(<span class="sc">~</span>assay) <span class="sc">+</span></span>
<span id="cb64-18"><a href="#cb64-18" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">70</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb64-19"><a href="#cb64-19" tabindex="-1"></a>            <span class="fu">ggtitle</span>(protName)</span>
<span id="cb64-20"><a href="#cb64-20" tabindex="-1"></a>        <span class="fu">print</span>(p1)</span>
<span id="cb64-21"><a href="#cb64-21" tabindex="-1"></a></span>
<span id="cb64-22"><a href="#cb64-22" tabindex="-1"></a>        <span class="co"># plotting 2</span></span>
<span id="cb64-23"><a href="#cb64-23" tabindex="-1"></a>        p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pePlotDf, <span class="fu">aes</span>(<span class="at">x =</span> colname, <span class="at">y =</span> value, <span class="at">fill =</span> genotype)) <span class="sc">+</span></span>
<span id="cb64-24"><a href="#cb64-24" tabindex="-1"></a>            <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb64-25"><a href="#cb64-25" tabindex="-1"></a>            <span class="fu">geom_point</span>(</span>
<span id="cb64-26"><a href="#cb64-26" tabindex="-1"></a>                <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> .<span class="dv">1</span>),</span>
<span id="cb64-27"><a href="#cb64-27" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">shape =</span> rowname)</span>
<span id="cb64-28"><a href="#cb64-28" tabindex="-1"></a>            ) <span class="sc">+</span></span>
<span id="cb64-29"><a href="#cb64-29" tabindex="-1"></a>            <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pePlotDf)) <span class="sc">+</span></span>
<span id="cb64-30"><a href="#cb64-30" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title =</span> protName, <span class="at">x =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">y =</span> <span class="st">&quot;peptide intensity (log2)&quot;</span>) <span class="sc">+</span></span>
<span id="cb64-31"><a href="#cb64-31" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">70</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb64-32"><a href="#cb64-32" tabindex="-1"></a>            <span class="fu">facet_grid</span>(<span class="sc">~</span>assay)</span>
<span id="cb64-33"><a href="#cb64-33" tabindex="-1"></a>        <span class="fu">print</span>(p2)</span>
<span id="cb64-34"><a href="#cb64-34" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-1.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-2.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-3.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-4.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-5.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-6.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-7.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-8.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-9.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-40-10.png" width="672" /></p>
</div>
</div>
<div id="experimental-design" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Experimental
Design</h1>
<div id="sample-size" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Sample size</h2>
<iframe width="560" height="315" src="https://www.youtube.com/embed/uELgkzDjVRY" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p><span class="math display">\[
\log_2 \text{FC} = \bar{y}_{p1}-\bar{y}_{p2}
\]</span></p>
<p><span class="math display">\[
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
\]</span></p>
<p><span class="math display">\[
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
\]</span></p>
<p>If we can assume equal variance in both treatment groups:</p>
<p><span class="math display">\[
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
\]</span></p>
<p><span class="math inline">\(\rightarrow\)</span> if number of
bio-repeats increases we have a higher power!</p>
<ul>
<li>cfr. Study of tamoxifen treated Estrogen Recepter (ER) positive
breast cancer patients</li>
</ul>
</div>
<div id="randomized-complete-block-designs" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Randomized complete
block designs</h2>
<iframe width="560" height="315" src="https://www.youtube.com/embed/HE4KWfIOVp0" frameborder="0" style="display: block; margin: auto;" allow="autoplay; encrypted-media" allowfullscreen>
</iframe>
<p><span class="math display">\[\sigma^2=
\sigma^2_{bio}+\sigma^2_\text{lab} +\sigma^2_\text{extraction} +
\sigma^2_\text{run} + \ldots\]</span></p>
<ul>
<li>Biological: fluctuations in protein level between mice, fluctations
in protein level between cells, …</li>
<li>Technical: cage effect, lab effect, week effect, plasma extraction,
MS-run, …</li>
</ul>
<div id="nature-methods-points-of-significance---blocking"
class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Nature methods:
Points of significance - Blocking</h3>
<p><a
href="https://www.nature.com/articles/nmeth.3005.pdf">https://www.nature.com/articles/nmeth.3005.pdf</a></p>
</div>
<div id="mouse-example" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Mouse example</h3>
<p><img src="figures/mouseTcell_RCB_design.png" width="50%" /></p>
<p>Duguet et al. (2017) MCP 16(8):1416-1432. doi:
10.1074/mcp.m116.062745</p>
<ul>
<li>All treatments of interest are present within block!</li>
<li>We can estimate the effect of the treatment within block!</li>
</ul>
<p>To illustrate the power of blocking we have subsetted the data of
Duguet et al. in a</p>
<ul>
<li><p>completely randomized design with</p>
<ul>
<li>four mice for which we only have measurements on the ordinary
T-cells</li>
<li>four mice for which we only have measurements on the regulatory
T-cells</li>
</ul></li>
<li><p>randomized complete block design with four mice for which we both
have</p>
<ul>
<li>measurements on ordinary T-cells as well as</li>
<li>measurements on regulatory T-cells</li>
</ul></li>
</ul>
</div>
<div id="data" class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Data</h3>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" tabindex="-1"></a>peptidesTable <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesRCB.txt&quot;</span>)</span>
<span id="cb65-2"><a href="#cb65-2" tabindex="-1"></a>int64 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">sapply</span>(peptidesTable,class) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>)</span>
<span id="cb65-3"><a href="#cb65-3" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> int64) peptidesTable[[j]] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(peptidesTable[[j]]) </span>
<span id="cb65-4"><a href="#cb65-4" tabindex="-1"></a></span>
<span id="cb65-5"><a href="#cb65-5" tabindex="-1"></a>peptidesTable2 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesCRD.txt&quot;</span>)</span>
<span id="cb65-6"><a href="#cb65-6" tabindex="-1"></a>int64 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">sapply</span>(peptidesTable2,class) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>)</span>
<span id="cb65-7"><a href="#cb65-7" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> int64) peptidesTable2[[j]] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(peptidesTable2[[j]]) </span>
<span id="cb65-8"><a href="#cb65-8" tabindex="-1"></a></span>
<span id="cb65-9"><a href="#cb65-9" tabindex="-1"></a>peptidesTable3 <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptides.txt&quot;</span>)</span>
<span id="cb65-10"><a href="#cb65-10" tabindex="-1"></a>int64 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">sapply</span>(peptidesTable3,class) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>)</span>
<span id="cb65-11"><a href="#cb65-11" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> int64) peptidesTable3[[j]] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(peptidesTable3[[j]]) </span>
<span id="cb65-12"><a href="#cb65-12" tabindex="-1"></a></span>
<span id="cb65-13"><a href="#cb65-13" tabindex="-1"></a>quantCols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity &quot;</span>, <span class="fu">names</span>(peptidesTable))</span>
<span id="cb65-14"><a href="#cb65-14" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb65-15"><a href="#cb65-15" tabindex="-1"></a>  <span class="at">assayData =</span> peptidesTable,</span>
<span id="cb65-16"><a href="#cb65-16" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb65-17"><a href="#cb65-17" tabindex="-1"></a>  <span class="at">quantCols =</span>  quantCols,</span>
<span id="cb65-18"><a href="#cb65-18" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb65-19"><a href="#cb65-19" tabindex="-1"></a><span class="fu">rm</span>(peptidesTable)</span>
<span id="cb65-20"><a href="#cb65-20" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8800234 470.0   15504024 828.1         NA 15504024 828.1
## Vcells 45274003 345.5   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8800228 470.0   15504024 828.1         NA 15504024 828.1
## Vcells 45271872 345.4   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" tabindex="-1"></a>quantCols2 <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity &quot;</span>, <span class="fu">names</span>(peptidesTable2))</span>
<span id="cb69-2"><a href="#cb69-2" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb69-3"><a href="#cb69-3" tabindex="-1"></a>  <span class="at">assayData =</span> peptidesTable2,</span>
<span id="cb69-4"><a href="#cb69-4" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb69-5"><a href="#cb69-5" tabindex="-1"></a>  <span class="at">quantCols =</span>  quantCols2,</span>
<span id="cb69-6"><a href="#cb69-6" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb69-7"><a href="#cb69-7" tabindex="-1"></a><span class="fu">rm</span>(peptidesTable2)</span>
<span id="cb69-8"><a href="#cb69-8" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8712110 465.3   15504024 828.1         NA 15504024 828.1
## Vcells 38667053 295.1   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8712109 465.3   15504024 828.1         NA 15504024 828.1
## Vcells 38662770 295.0   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" tabindex="-1"></a>quantCols3 <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity &quot;</span>, <span class="fu">names</span>(peptidesTable3))</span>
<span id="cb73-2"><a href="#cb73-2" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb73-3"><a href="#cb73-3" tabindex="-1"></a>  <span class="at">assayData =</span> peptidesTable3,</span>
<span id="cb73-4"><a href="#cb73-4" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb73-5"><a href="#cb73-5" tabindex="-1"></a>  <span class="at">quantCols =</span>  quantCols3,</span>
<span id="cb73-6"><a href="#cb73-6" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span>
<span id="cb73-7"><a href="#cb73-7" tabindex="-1"></a><span class="fu">rm</span>(peptidesTable3)</span>
<span id="cb73-8"><a href="#cb73-8" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8711698 465.3   15504024 828.1         NA 15504024 828.1
## Vcells 29978673 228.8   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8711623 465.3   15504024 828.1         NA 15504024 828.1
## Vcells 29974115 228.7   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" tabindex="-1"></a><span class="do">### Design</span></span>
<span id="cb77-2"><a href="#cb77-2" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb77-3"><a href="#cb77-3" tabindex="-1"></a>  <span class="fu">colnames</span>(pe[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb77-4"><a href="#cb77-4" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb77-5"><a href="#cb77-5" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">|&gt;</span></span>
<span id="cb77-6"><a href="#cb77-6" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">|&gt;</span>  </span>
<span id="cb77-7"><a href="#cb77-7" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb77-8"><a href="#cb77-8" tabindex="-1"></a></span>
<span id="cb77-9"><a href="#cb77-9" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe[[<span class="dv">1</span>]] <span class="sc">|&gt;</span></span>
<span id="cb77-10"><a href="#cb77-10" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span></span>
<span id="cb77-11"><a href="#cb77-11" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[ ]&quot;</span>)  <span class="sc">|&gt;</span></span>
<span id="cb77-12"><a href="#cb77-12" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">|&gt;</span></span>
<span id="cb77-13"><a href="#cb77-13" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb77-14"><a href="#cb77-14" tabindex="-1"></a></span>
<span id="cb77-15"><a href="#cb77-15" tabindex="-1"></a><span class="fu">colData</span>(pe2)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb77-16"><a href="#cb77-16" tabindex="-1"></a>  <span class="fu">colnames</span>(pe2[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb77-17"><a href="#cb77-17" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb77-18"><a href="#cb77-18" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">|&gt;</span></span>
<span id="cb77-19"><a href="#cb77-19" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">|&gt;</span>  </span>
<span id="cb77-20"><a href="#cb77-20" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb77-21"><a href="#cb77-21" tabindex="-1"></a></span>
<span id="cb77-22"><a href="#cb77-22" tabindex="-1"></a><span class="fu">colData</span>(pe2)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe2[[<span class="dv">1</span>]] <span class="sc">|&gt;</span></span>
<span id="cb77-23"><a href="#cb77-23" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span></span>
<span id="cb77-24"><a href="#cb77-24" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[ ]&quot;</span>)  <span class="sc">|&gt;</span></span>
<span id="cb77-25"><a href="#cb77-25" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">|&gt;</span></span>
<span id="cb77-26"><a href="#cb77-26" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb77-27"><a href="#cb77-27" tabindex="-1"></a></span>
<span id="cb77-28"><a href="#cb77-28" tabindex="-1"></a><span class="fu">colData</span>(pe3)<span class="sc">$</span>celltype <span class="ot">&lt;-</span> <span class="fu">substr</span>(</span>
<span id="cb77-29"><a href="#cb77-29" tabindex="-1"></a>  <span class="fu">colnames</span>(pe3[[<span class="st">&quot;peptideRaw&quot;</span>]]),</span>
<span id="cb77-30"><a href="#cb77-30" tabindex="-1"></a>  <span class="dv">11</span>,</span>
<span id="cb77-31"><a href="#cb77-31" tabindex="-1"></a>  <span class="dv">14</span>) <span class="sc">|&gt;</span></span>
<span id="cb77-32"><a href="#cb77-32" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">|&gt;</span>  </span>
<span id="cb77-33"><a href="#cb77-33" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span>
<span id="cb77-34"><a href="#cb77-34" tabindex="-1"></a></span>
<span id="cb77-35"><a href="#cb77-35" tabindex="-1"></a><span class="fu">colData</span>(pe3)<span class="sc">$</span>mouse <span class="ot">&lt;-</span> pe3[[<span class="dv">1</span>]] <span class="sc">|&gt;</span></span>
<span id="cb77-36"><a href="#cb77-36" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span></span>
<span id="cb77-37"><a href="#cb77-37" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="at">split=</span><span class="st">&quot;[ ]&quot;</span>)  <span class="sc">|&gt;</span></span>
<span id="cb77-38"><a href="#cb77-38" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) x[<span class="dv">3</span>]) <span class="sc">|&gt;</span></span>
<span id="cb77-39"><a href="#cb77-39" tabindex="-1"></a>  <span class="fu">as.factor</span>()</span></code></pre></div>
</p>
<p></design></p>
</div>
<div id="preprocessing-1" class="section level3" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Preprocessing</h3>
<div id="log-transform" class="section level4" number="2.2.4.1">
<h4><span class="header-section-number">2.2.4.1</span>
Log-transform</h4>
<details>
<summary>
Click to see code to log-transfrom the data
</summary>
<p>
<ul>
<li>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb78-2"><a href="#cb78-2" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe2, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span>
<span id="cb78-4"><a href="#cb78-4" tabindex="-1"></a></span>
<span id="cb78-5"><a href="#cb78-5" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe3, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb79-2"><a href="#cb79-2" tabindex="-1"></a></span>
<span id="cb79-3"><a href="#cb79-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe2, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb79-4"><a href="#cb79-4" tabindex="-1"></a></span>
<span id="cb79-5"><a href="#cb79-5" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe3, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="filtering" class="section level4" number="2.2.4.2">
<h4><span class="header-section-number">2.2.4.2</span> Filtering</h4>
<details>
<summary>
Click to see details on filtering
</summary>
<p>
<ol style="list-style-type: decimal">
<li>Remove peptides that map to multiple proteins</li>
</ol>
<p>We remove PSMs that could not be mapped to a protein or that map to
multiple proteins (the protein identifier contains multiple identifiers
separated by a <code>;</code>).</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb80-2"><a href="#cb80-2" tabindex="-1"></a>    pe, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb80-3"><a href="#cb80-3" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;;&quot;</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb82-2"><a href="#cb82-2" tabindex="-1"></a>    pe2, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb82-3"><a href="#cb82-3" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;;&quot;</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb84-2"><a href="#cb84-2" tabindex="-1"></a>    pe3, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb84-3"><a href="#cb84-3" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;;&quot;</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants, peptides that map to decoy sequences,
and proteins which were only identified by peptides with
modifications.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe2,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe3,<span class="sc">~</span> Potential.contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Potential.contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were identified in less than three sample</li>
</ol>
<p>We keep peptides that were observed at least three times.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb98-2"><a href="#cb98-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span>
<span id="cb98-3"><a href="#cb98-3" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb98-4"><a href="#cb98-4" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb98-5"><a href="#cb98-5" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 38782</code></pre>
<div class="sourceCode" id="cb100"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe2[[<span class="st">&quot;peptideLog&quot;</span>]])</span>
<span id="cb100-2"><a href="#cb100-2" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb100-3"><a href="#cb100-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe2, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb100-4"><a href="#cb100-4" tabindex="-1"></a><span class="fu">nrow</span>(pe2[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 37898</code></pre>
<div class="sourceCode" id="cb102"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe3[[<span class="st">&quot;peptideLog&quot;</span>]])</span>
<span id="cb102-2"><a href="#cb102-2" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb102-3"><a href="#cb102-3" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe3, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb102-4"><a href="#cb102-4" tabindex="-1"></a><span class="fu">nrow</span>(pe3[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 43984</code></pre>
</p>
</details>
</div>
<div id="normalization" class="section level4" number="2.2.4.3">
<h4><span class="header-section-number">2.2.4.3</span>
Normalization</h4>
<details>
<summary>
Click to see code to normalize the data
</summary>
<p>
<div class="sourceCode" id="cb104"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb104-2"><a href="#cb104-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb104-3"><a href="#cb104-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb104-4"><a href="#cb104-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb104-5"><a href="#cb104-5" tabindex="-1"></a></span>
<span id="cb104-6"><a href="#cb104-6" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe2, </span>
<span id="cb104-7"><a href="#cb104-7" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb104-8"><a href="#cb104-8" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb104-9"><a href="#cb104-9" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span>
<span id="cb104-10"><a href="#cb104-10" tabindex="-1"></a></span>
<span id="cb104-11"><a href="#cb104-11" tabindex="-1"></a></span>
<span id="cb104-12"><a href="#cb104-12" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe3, </span>
<span id="cb104-13"><a href="#cb104-13" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb104-14"><a href="#cb104-14" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb104-15"><a href="#cb104-15" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
<div id="summarization" class="section level4" number="2.2.4.4">
<h4><span class="header-section-number">2.2.4.4</span>
Summarization</h4>
<details>
<summary>
Click to see code to summarize the data
</summary>
<p>
<div class="sourceCode" id="cb105"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb105-2"><a href="#cb105-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb105-3"><a href="#cb105-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb105-4"><a href="#cb105-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-5"><a href="#cb105-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb108"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe2,</span>
<span id="cb108-2"><a href="#cb108-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb108-3"><a href="#cb108-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb108-4"><a href="#cb108-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-5"><a href="#cb108-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.
## Aggregated: 1/1</code></pre>
<div class="sourceCode" id="cb110"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe3,</span>
<span id="cb110-2"><a href="#cb110-2" tabindex="-1"></a> <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>,</span>
<span id="cb110-3"><a href="#cb110-3" tabindex="-1"></a> <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>,</span>
<span id="cb110-4"><a href="#cb110-4" tabindex="-1"></a> <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb110-5"><a href="#cb110-5" tabindex="-1"></a> <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.
## Aggregated: 1/1</code></pre>
</div>
<div id="filtering-proteins-with-too-many-missing-values"
class="section level4" number="2.2.4.5">
<h4><span class="header-section-number">2.2.4.5</span> Filtering
proteins with too many missing values</h4>
<p>We want to have at least two observed protein intensities for each
group so we set the minimum number of observed values at 4. We still
have to check for the observed proteins if that is the case.</p>
<p>For block design more clever filtering can be used. E.g. we could
imply that we have both cell types in at least two animals…</p>
<div class="sourceCode" id="cb112"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb112-2"><a href="#cb112-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="st">&quot;protein&quot;</span>]])</span>
<span id="cb112-3"><a href="#cb112-3" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb112-4"><a href="#cb112-4" tabindex="-1"></a>pe<span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>)</span>
<span id="cb112-5"><a href="#cb112-5" tabindex="-1"></a></span>
<span id="cb112-6"><a href="#cb112-6" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])</span>
<span id="cb112-7"><a href="#cb112-7" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb112-8"><a href="#cb112-8" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe2, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>)</span>
<span id="cb112-9"><a href="#cb112-9" tabindex="-1"></a></span>
<span id="cb112-10"><a href="#cb112-10" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe3[[<span class="st">&quot;protein&quot;</span>]])</span>
<span id="cb112-11"><a href="#cb112-11" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb112-12"><a href="#cb112-12" tabindex="-1"></a>pe3 <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe3, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
</p>
</details>
</div>
</div>
<div id="data-exploration-what-is-impact-of-blocking"
class="section level3" number="2.2.5">
<h3><span class="header-section-number">2.2.5</span> Data Exploration:
what is impact of blocking?</h3>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb113"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe3)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>)</span>
<span id="cb113-2"><a href="#cb113-2" tabindex="-1"></a>mdsObj3 <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe3[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-3"><a href="#cb113-3" tabindex="-1"></a>mdsOrig <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe3) <span class="sc">|&gt;</span></span>
<span id="cb113-4"><a href="#cb113-4" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb113-5"><a href="#cb113-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj3<span class="sc">$</span>x,</span>
<span id="cb113-6"><a href="#cb113-6" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj3<span class="sc">$</span>y,</span>
<span id="cb113-7"><a href="#cb113-7" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb113-8"><a href="#cb113-8" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb113-9"><a href="#cb113-9" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb113-10"><a href="#cb113-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb113-11"><a href="#cb113-11" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb113-12"><a href="#cb113-12" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb113-13"><a href="#cb113-13" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-14"><a href="#cb113-14" tabindex="-1"></a>      mdsObj3<span class="sc">$</span>axislabel,</span>
<span id="cb113-15"><a href="#cb113-15" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-16"><a href="#cb113-16" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb113-17"><a href="#cb113-17" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-18"><a href="#cb113-18" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-19"><a href="#cb113-19" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj3<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-20"><a href="#cb113-20" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-21"><a href="#cb113-21" tabindex="-1"></a>        ),</span>
<span id="cb113-22"><a href="#cb113-22" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-23"><a href="#cb113-23" tabindex="-1"></a>      )</span>
<span id="cb113-24"><a href="#cb113-24" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-25"><a href="#cb113-25" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb113-26"><a href="#cb113-26" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-27"><a href="#cb113-27" tabindex="-1"></a>      mdsObj3<span class="sc">$</span>axislabel,</span>
<span id="cb113-28"><a href="#cb113-28" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-29"><a href="#cb113-29" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb113-30"><a href="#cb113-30" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-31"><a href="#cb113-31" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-32"><a href="#cb113-32" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj3<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-33"><a href="#cb113-33" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-34"><a href="#cb113-34" tabindex="-1"></a>        ),</span>
<span id="cb113-35"><a href="#cb113-35" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-36"><a href="#cb113-36" tabindex="-1"></a>      )</span>
<span id="cb113-37"><a href="#cb113-37" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-38"><a href="#cb113-38" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Original (RCB)&quot;</span>)</span>
<span id="cb113-39"><a href="#cb113-39" tabindex="-1"></a></span>
<span id="cb113-40"><a href="#cb113-40" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb113-41"><a href="#cb113-41" tabindex="-1"></a>mdsObj <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-42"><a href="#cb113-42" tabindex="-1"></a>mdsRCB <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe) <span class="sc">|&gt;</span></span>
<span id="cb113-43"><a href="#cb113-43" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb113-44"><a href="#cb113-44" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj<span class="sc">$</span>x,</span>
<span id="cb113-45"><a href="#cb113-45" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj<span class="sc">$</span>y,</span>
<span id="cb113-46"><a href="#cb113-46" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb113-47"><a href="#cb113-47" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb113-48"><a href="#cb113-48" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb113-49"><a href="#cb113-49" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb113-50"><a href="#cb113-50" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb113-51"><a href="#cb113-51" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb113-52"><a href="#cb113-52" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-53"><a href="#cb113-53" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb113-54"><a href="#cb113-54" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-55"><a href="#cb113-55" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb113-56"><a href="#cb113-56" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-57"><a href="#cb113-57" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-58"><a href="#cb113-58" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-59"><a href="#cb113-59" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-60"><a href="#cb113-60" tabindex="-1"></a>        ),</span>
<span id="cb113-61"><a href="#cb113-61" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-62"><a href="#cb113-62" tabindex="-1"></a>      )</span>
<span id="cb113-63"><a href="#cb113-63" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-64"><a href="#cb113-64" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb113-65"><a href="#cb113-65" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-66"><a href="#cb113-66" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb113-67"><a href="#cb113-67" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-68"><a href="#cb113-68" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb113-69"><a href="#cb113-69" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-70"><a href="#cb113-70" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-71"><a href="#cb113-71" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-72"><a href="#cb113-72" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-73"><a href="#cb113-73" tabindex="-1"></a>        ),</span>
<span id="cb113-74"><a href="#cb113-74" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-75"><a href="#cb113-75" tabindex="-1"></a>      )</span>
<span id="cb113-76"><a href="#cb113-76" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-77"><a href="#cb113-77" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Randomized Complete Block (RCB)&quot;</span>)</span>
<span id="cb113-78"><a href="#cb113-78" tabindex="-1"></a></span>
<span id="cb113-79"><a href="#cb113-79" tabindex="-1"></a></span>
<span id="cb113-80"><a href="#cb113-80" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(pe2)<span class="sc">$</span>mouse) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;m&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)</span>
<span id="cb113-81"><a href="#cb113-81" tabindex="-1"></a>mdsObj2 <span class="ot">&lt;-</span> <span class="fu">plotMDS</span>(<span class="fu">assay</span>(pe2[[<span class="st">&quot;protein&quot;</span>]]), <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb113-82"><a href="#cb113-82" tabindex="-1"></a>mdsCRD <span class="ot">&lt;-</span> <span class="fu">colData</span>(pe2) <span class="sc">|&gt;</span></span>
<span id="cb113-83"><a href="#cb113-83" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb113-84"><a href="#cb113-84" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mds1 =</span> mdsObj2<span class="sc">$</span>x,</span>
<span id="cb113-85"><a href="#cb113-85" tabindex="-1"></a>         <span class="at">mds2 =</span> mdsObj2<span class="sc">$</span>y,</span>
<span id="cb113-86"><a href="#cb113-86" tabindex="-1"></a>         <span class="at">lab =</span> <span class="fu">paste</span>(mouse,celltype,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb113-87"><a href="#cb113-87" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mds1, <span class="at">y =</span> mds2, <span class="at">label =</span> lab, <span class="at">color =</span> celltype, <span class="at">group =</span> mouse)) <span class="sc">+</span></span>
<span id="cb113-88"><a href="#cb113-88" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb113-89"><a href="#cb113-89" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb113-90"><a href="#cb113-90" tabindex="-1"></a>  <span class="fu">xlab</span>(</span>
<span id="cb113-91"><a href="#cb113-91" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-92"><a href="#cb113-92" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb113-93"><a href="#cb113-93" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-94"><a href="#cb113-94" tabindex="-1"></a>      <span class="dv">1</span>, </span>
<span id="cb113-95"><a href="#cb113-95" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-96"><a href="#cb113-96" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-97"><a href="#cb113-97" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj2<span class="sc">$</span>var.explained[<span class="dv">1</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-98"><a href="#cb113-98" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-99"><a href="#cb113-99" tabindex="-1"></a>        ),</span>
<span id="cb113-100"><a href="#cb113-100" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-101"><a href="#cb113-101" tabindex="-1"></a>      )</span>
<span id="cb113-102"><a href="#cb113-102" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-103"><a href="#cb113-103" tabindex="-1"></a>  <span class="fu">ylab</span>(</span>
<span id="cb113-104"><a href="#cb113-104" tabindex="-1"></a>    <span class="fu">paste0</span>(</span>
<span id="cb113-105"><a href="#cb113-105" tabindex="-1"></a>      mdsObj<span class="sc">$</span>axislabel,</span>
<span id="cb113-106"><a href="#cb113-106" tabindex="-1"></a>      <span class="st">&quot; &quot;</span>,</span>
<span id="cb113-107"><a href="#cb113-107" tabindex="-1"></a>      <span class="dv">2</span>, </span>
<span id="cb113-108"><a href="#cb113-108" tabindex="-1"></a>      <span class="st">&quot; (&quot;</span>,</span>
<span id="cb113-109"><a href="#cb113-109" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb113-110"><a href="#cb113-110" tabindex="-1"></a>        <span class="fu">round</span>(mdsObj2<span class="sc">$</span>var.explained[<span class="dv">2</span>] <span class="sc">*</span><span class="dv">100</span>,<span class="dv">0</span>),</span>
<span id="cb113-111"><a href="#cb113-111" tabindex="-1"></a>        <span class="st">&quot;%&quot;</span></span>
<span id="cb113-112"><a href="#cb113-112" tabindex="-1"></a>        ),</span>
<span id="cb113-113"><a href="#cb113-113" tabindex="-1"></a>      <span class="st">&quot;)&quot;</span></span>
<span id="cb113-114"><a href="#cb113-114" tabindex="-1"></a>      )</span>
<span id="cb113-115"><a href="#cb113-115" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb113-116"><a href="#cb113-116" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Completely Randomized Design (CRD)&quot;</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb114"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" tabindex="-1"></a>mdsOrig</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-52-1.png" width="672" /></p>
<div class="sourceCode" id="cb115"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" tabindex="-1"></a>mdsRCB</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-52-2.png" width="672" /></p>
<div class="sourceCode" id="cb116"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" tabindex="-1"></a>mdsCRD</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-52-3.png" width="672" /></p>
<ul>
<li><p>We observe that the leading fold change is according to
mouse</p></li>
<li><p>In the second dimension we see a separation according to
cell-type</p></li>
<li><p>With the Randomized Complete Block design (RCB) we can remove the
mouse effect from the analysis!</p></li>
<li><p>We can isolate the between block variability from the analysis
using linear model:</p>
<ul>
<li><p>Formula in R <span class="math display">\[
y \sim \text{celltype} + \text{mouse}
\]</span></p></li>
<li><p>Formula</p></li>
</ul></li>
</ul>
<p><span class="math display">\[
y_i = \beta_0 + \beta_\text{Treg} x_{i,\text{Treg}} + \beta_{m2}x_{i,m2}
+ \beta_{m3}x_{i,m3} + \beta_{m4}x_{i,m4}  + \epsilon_i
\]</span></p>
<p>with</p>
<ul>
<li><p><span class="math inline">\(x_{i,Treg}=\begin{cases}
1&amp; \text{Treg}\\
0&amp; \text{Tcon}
\end{cases}\)</span></p></li>
<li><p><span class="math inline">\(x_{i,m2}=\begin{cases}
1&amp; \text{m2}\\
0&amp; \text{otherwise}
\end{cases}\)</span></p></li>
<li><p><span class="math inline">\(x_{i,m3}=\begin{cases}
1&amp; \text{m3}\\
0&amp; \text{otherwise}
\end{cases}\)</span></p></li>
<li><p><span class="math inline">\(x_{i,m4}=\begin{cases}
1&amp; \text{m4}\\
0&amp; \text{otherwise}
\end{cases}\)</span></p></li>
<li><p>Possible in msqrob2 and MSstats but not possible with
Perseus!</p></li>
</ul>
</div>
<div id="modeling-and-inference" class="section level3" number="2.2.6">
<h3><span class="header-section-number">2.2.6</span> Modeling and
inference</h3>
<div id="rcb-analysis" class="section level4" number="2.2.6.1">
<h4><span class="header-section-number">2.2.6.1</span> RCB analysis</h4>
<div class="sourceCode" id="cb117"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb117-2"><a href="#cb117-2" tabindex="-1"></a>  <span class="at">object =</span> pe,</span>
<span id="cb117-3"><a href="#cb117-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb117-4"><a href="#cb117-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> celltype <span class="sc">+</span> mouse)</span></code></pre></div>
</div>
<div id="crd-analysis" class="section level4" number="2.2.6.2">
<h4><span class="header-section-number">2.2.6.2</span> CRD analysis</h4>
<div class="sourceCode" id="cb118"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb118-2"><a href="#cb118-2" tabindex="-1"></a>  <span class="at">object =</span> pe2,</span>
<span id="cb118-3"><a href="#cb118-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb118-4"><a href="#cb118-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> celltype)</span></code></pre></div>
</div>
<div id="estimation-effect-size-and-inference" class="section level4"
number="2.2.6.3">
<h4><span class="header-section-number">2.2.6.3</span> Estimation,
effect size and inference</h4>
<p>Effect size in RCB</p>
<div class="sourceCode" id="cb119"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" tabindex="-1"></a><span class="fu">library</span>(ExploreModelMatrix)</span>
<span id="cb119-2"><a href="#cb119-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> celltype <span class="sc">+</span> mouse)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-55-1.png" width="672" /></p>
<p>Effect size in CRD</p>
<div class="sourceCode" id="cb121"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe2),<span class="sc">~</span> celltype)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-56-1.png" width="672" /></p>
<details>
<summary>
Click to see code for statistical inference
</summary>
<p>
<div class="sourceCode" id="cb123"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;celltypeTreg = 0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;celltypeTreg&quot;</span>))</span>
<span id="cb123-2"><a href="#cb123-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span>
<span id="cb123-3"><a href="#cb123-3" tabindex="-1"></a>pe2 <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe2, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
</p>
</details>
</div>
</div>
<div id="comparison-of-results" class="section level3" number="2.2.7">
<h3><span class="header-section-number">2.2.7</span> Comparison of
results</h3>
<details>
<summary>
Click to see code
</summary>
<p>
</p>
</details>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-59-1.png" width="672" /><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-59-2.png" width="672" /></p>
</div>
<div id="comparison-of-standard-deviation" class="section level3"
number="2.2.8">
<h3><span class="header-section-number">2.2.8</span> Comparison of
standard deviation</h3>
<details>
<summary>
Click to see code
</summary>
<p>
<div class="sourceCode" id="cb124"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" tabindex="-1"></a>accessions <span class="ot">&lt;-</span> <span class="fu">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])[<span class="fu">rownames</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">%in%</span><span class="fu">rownames</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])]</span>
<span id="cb124-2"><a href="#cb124-2" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb124-3"><a href="#cb124-3" tabindex="-1"></a><span class="at">sigmaRBC =</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[accessions], getSigmaPosterior),</span>
<span id="cb124-4"><a href="#cb124-4" tabindex="-1"></a>sigmaCRD <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">rowData</span>(pe2[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>msqrobModels[accessions], getSigmaPosterior)</span>
<span id="cb124-5"><a href="#cb124-5" tabindex="-1"></a>)</span>
<span id="cb124-6"><a href="#cb124-6" tabindex="-1"></a></span>
<span id="cb124-7"><a href="#cb124-7" tabindex="-1"></a>plotRBCvsCRD <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> dat, <span class="fu">aes</span>(sigmaRBC, sigmaCRD)) <span class="sc">+</span></span>
<span id="cb124-8"><a href="#cb124-8" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">shape =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb124-9"><a href="#cb124-9" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb124-10"><a href="#cb124-10" tabindex="-1"></a>    <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb124-11"><a href="#cb124-11" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>,<span class="at">slope=</span><span class="dv">1</span>)</span></code></pre></div>
</p>
</details>
<div class="sourceCode" id="cb125"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" tabindex="-1"></a>  plotRBCvsCRD</span></code></pre></div>
<pre><code>## Warning: Removed 351 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<ul>
<li><p>We clearly observe that the standard deviation of the protein
expression in the RCB is smaller for the majority of the proteins than
that obtained with the CRD</p></li>
<li><p>Can you think of a reason why it would not be useful to block on
a particular factor?</p></li>
</ul>
</div>
</div>
<div id="pseudo-replication" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span>
Pseudo-replication</h2>
<ul>
<li>The Francisella tularensis that we used before was a subset of the
data of Ramond et al. </li>
<li>The authors have run the sample for each bio-rep in technical
triplicate on the mass-spectrometer.</li>
</ul>
<div id="data-1" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Data</h3>
<details>
<summary>
Click to see code to import data
</summary>
<p>
<ol style="list-style-type: decimal">
<li>We use a peptides.txt file from MS-data quantified with maxquant
that contains MS1 intensities summarized at the peptide level.</li>
</ol>
<div class="sourceCode" id="cb127"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" tabindex="-1"></a>peptidesTable <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/refs/heads/master/Francisella/data/maxquant/peptides.txt&quot;</span>)</span>
<span id="cb127-2"><a href="#cb127-2" tabindex="-1"></a>int64 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">sapply</span>(peptidesTable,class) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>)</span>
<span id="cb127-3"><a href="#cb127-3" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> int64) peptidesTable[[j]] <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(peptidesTable[[j]])                      </span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Maxquant stores the intensity data for the different samples in
columnns that start with Intensity. We can retreive the column names
with the intensity data with the code below:</li>
</ol>
<div class="sourceCode" id="cb128"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" tabindex="-1"></a>quantCols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;Intensity &quot;</span>, <span class="fu">names</span>(peptidesTable))</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Read the data and store it in QFeatures object</li>
</ol>
<div class="sourceCode" id="cb129"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb129-2"><a href="#cb129-2" tabindex="-1"></a>  <span class="at">assayData =</span> peptidesTable,</span>
<span id="cb129-3"><a href="#cb129-3" tabindex="-1"></a>  <span class="at">fnames =</span> <span class="dv">1</span>,</span>
<span id="cb129-4"><a href="#cb129-4" tabindex="-1"></a>  <span class="at">quantCols =</span>  quantCols,</span>
<span id="cb129-5"><a href="#cb129-5" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;peptideRaw&quot;</span>)</span></code></pre></div>
<pre><code>## Checking arguments.</code></pre>
<pre><code>## Loading data as a &#39;SummarizedExperiment&#39; object.</code></pre>
<pre><code>## Formatting sample annotations (colData).</code></pre>
<pre><code>## Formatting data as a &#39;QFeatures&#39; object.</code></pre>
<pre><code>## Setting assay rownames.</code></pre>
<div class="sourceCode" id="cb135"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" tabindex="-1"></a><span class="fu">rm</span>(peptidesTable)</span>
<span id="cb135-2"><a href="#cb135-2" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8796284 469.8   15504024 828.1         NA 15504024 828.1
## Vcells 34766774 265.3   81347764 620.7      16384 81347764 620.7</code></pre>
<div class="sourceCode" id="cb137"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" tabindex="-1"></a><span class="fu">gc</span>()</span></code></pre></div>
<pre><code>##            used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
## Ncells  8796277 469.8   15504024 828.1         NA 15504024 828.1
## Vcells 34766799 265.3   81347764 620.7      16384 81347764 620.7</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Update data with information on design</li>
</ol>
<div class="sourceCode" id="cb139"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>genotype <span class="ot">&lt;-</span> pe[[<span class="dv">1</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb139-2"><a href="#cb139-2" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span> </span>
<span id="cb139-3"><a href="#cb139-3" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="dv">12</span>,<span class="dv">13</span>) <span class="sc">|&gt;</span></span>
<span id="cb139-4"><a href="#cb139-4" tabindex="-1"></a>  <span class="fu">as.factor</span>() <span class="sc">|&gt;</span> </span>
<span id="cb139-5"><a href="#cb139-5" tabindex="-1"></a>  <span class="fu">relevel</span>(<span class="st">&quot;WT&quot;</span>)</span>
<span id="cb139-6"><a href="#cb139-6" tabindex="-1"></a></span>
<span id="cb139-7"><a href="#cb139-7" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>biorep <span class="ot">&lt;-</span> pe[[<span class="dv">1</span>]] <span class="sc">|&gt;</span> </span>
<span id="cb139-8"><a href="#cb139-8" tabindex="-1"></a>  <span class="fu">colnames</span>() <span class="sc">|&gt;</span> </span>
<span id="cb139-9"><a href="#cb139-9" tabindex="-1"></a>  <span class="fu">substr</span>(<span class="dv">22</span>,<span class="dv">22</span>)</span>
<span id="cb139-10"><a href="#cb139-10" tabindex="-1"></a></span>
<span id="cb139-11"><a href="#cb139-11" tabindex="-1"></a><span class="fu">colData</span>(pe)<span class="sc">$</span>biorep <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">colData</span>(pe)<span class="sc">$</span>genotype,<span class="fu">colData</span>(pe)<span class="sc">$</span>biorep,<span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb139-12"><a href="#cb139-12" tabindex="-1"></a>  </span>
<span id="cb139-13"><a href="#cb139-13" tabindex="-1"></a>pe <span class="sc">|&gt;</span> <span class="fu">colData</span>()</span></code></pre></div>
<pre><code>## DataFrame with 18 rows and 2 columns
##                          genotype      biorep
##                          &lt;factor&gt; &lt;character&gt;
## Intensity 1WT_20_2h_n3_1       WT        WT_3
## Intensity 1WT_20_2h_n3_2       WT        WT_3
## Intensity 1WT_20_2h_n3_3       WT        WT_3
## Intensity 1WT_20_2h_n4_1       WT        WT_4
## Intensity 1WT_20_2h_n4_2       WT        WT_4
## ...                           ...         ...
## Intensity 3D8_20_2h_n4_2       D8        D8_4
## Intensity 3D8_20_2h_n4_3       D8        D8_4
## Intensity 3D8_20_2h_n5_1       D8        D8_5
## Intensity 3D8_20_2h_n5_2       D8        D8_5
## Intensity 3D8_20_2h_n5_3       D8        D8_5</code></pre>
</p>
</details>
</div>
<div id="preprocessing-2" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Preprocessing</h3>
<details>
<summary>
Click to see code to log-transfrom the data
</summary>
<p>
<ol style="list-style-type: decimal">
<li>Log transform</li>
</ol>
<ul>
<li>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code>.</li>
</ul>
<div class="sourceCode" id="cb141"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">&quot;peptideRaw&quot;</span>) <span class="co"># convert 0 to NA</span></span></code></pre></div>
<ul>
<li>Logtransform data with base 2</li>
</ul>
<div class="sourceCode" id="cb142"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">&quot;peptideRaw&quot;</span>, <span class="at">name =</span> <span class="st">&quot;peptideLog&quot;</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Filtering</li>
</ol>
<ul>
<li>We remove PSMs that could not be mapped to a protein or that map to
multiple proteins (the protein identifier contains multiple identifiers
separated by a <code>;</code>).</li>
</ul>
<div class="sourceCode" id="cb143"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb143-2"><a href="#cb143-2" tabindex="-1"></a>    pe, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">&quot;&quot;</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb143-3"><a href="#cb143-3" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;;&quot;</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<pre><code>## &#39;Proteins&#39; found in 2 out of 2 assay(s).</code></pre>
<ul>
<li>Remove reverse sequences (decoys) and contaminants. Note that this
is indicated by the column names Reverse and depending on the version of
maxQuant with Potential.contaminants or Contaminants.</li>
</ul>
<div class="sourceCode" id="cb145"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span>Reverse <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Reverse&#39; found in 2 out of 2 assay(s).</code></pre>
<div class="sourceCode" id="cb147"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,<span class="sc">~</span> Contaminant <span class="sc">!=</span> <span class="st">&quot;+&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;Contaminant&#39; found in 2 out of 2 assay(s).</code></pre>
<ul>
<li>Drop peptides that were identified in less than three sample. We
tolerate the following proportion of NAs: pNA = (n-3)/n. </li>
</ul>
<div class="sourceCode" id="cb149"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">3</span></span>
<span id="cb149-2"><a href="#cb149-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span>
<span id="cb149-3"><a href="#cb149-3" tabindex="-1"></a>pNA <span class="ot">&lt;-</span> (n<span class="sc">-</span>nObs)<span class="sc">/</span>n</span>
<span id="cb149-4"><a href="#cb149-4" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(pe, <span class="at">pNA =</span> pNA, <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>)</span>
<span id="cb149-5"><a href="#cb149-5" tabindex="-1"></a><span class="fu">nrow</span>(pe[[<span class="st">&quot;peptideLog&quot;</span>]])</span></code></pre></div>
<pre><code>## [1] 7244</code></pre>
<p>We keep 7244 peptides upon filtering.</p>
<ol start="3" style="list-style-type: decimal">
<li>Normalization by median centering</li>
</ol>
<div class="sourceCode" id="cb151"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(pe, </span>
<span id="cb151-2"><a href="#cb151-2" tabindex="-1"></a>                <span class="at">i =</span> <span class="st">&quot;peptideLog&quot;</span>, </span>
<span id="cb151-3"><a href="#cb151-3" tabindex="-1"></a>                <span class="at">name =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb151-4"><a href="#cb151-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;center.median&quot;</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Summarization. We use the standard sumarisation in
aggregateFeatures, which is a robust summarisation method.</li>
</ol>
<div class="sourceCode" id="cb152"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(pe,</span>
<span id="cb152-2"><a href="#cb152-2" tabindex="-1"></a>    <span class="at">i =</span> <span class="st">&quot;peptideNorm&quot;</span>, </span>
<span id="cb152-3"><a href="#cb152-3" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">&quot;Proteins&quot;</span>, </span>
<span id="cb152-4"><a href="#cb152-4" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb152-5"><a href="#cb152-5" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<pre><code>## Your quantitative and row data contain missing values. Please read the
## relevant section(s) in the aggregateFeatures manual page regarding the
## effects of missing values on data aggregation.</code></pre>
<pre><code>## Aggregated: 1/1</code></pre>
</p>
</details>
<div class="sourceCode" id="cb155"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" tabindex="-1"></a>msqrob2gui<span class="sc">:::</span><span class="fu">plotPCA</span>(pe, <span class="st">&quot;protein&quot;</span>, <span class="st">&quot;biorep&quot;</span>)</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-73-1.png" width="672" /></p>
<ul>
<li>Response?</li>
<li>Experimental unit?</li>
<li>Observational unit?</li>
<li>Factors?</li>
</ul>
<div class="sourceCode" id="cb156"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(<span class="fu">colData</span>(pe),<span class="sc">~</span> genotype <span class="sc">+</span> biorep)<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-74-1.png" width="672" /></p>
<p><span class="math inline">\(\rightarrow\)</span> Pseudo-replication,
randomisation to bio-repeat and each bio-repeat measured in technical
triplicate. <span class="math inline">\(\rightarrow\)</span> If we would
analyse the data using a linear model based on each measured intensity,
we would act as if we had sampled 18 bio-repeats. <span
class="math inline">\(\rightarrow\)</span> Effect of interest has to be
assessed between bio-repeats. So block analysis is not possible (if we
condition on bio-repeat, we condition on the genotype)!</p>
</div>
<div id="wrong-analysis" class="section level3" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Wrong analysis</h3>
<div class="sourceCode" id="cb158"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span>genotype, <span class="at">modelColumnName =</span> <span class="st">&quot;wrong&quot;</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-2"><a href="#cb158-2" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;genotypeD8 = 0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;genotypeD8&quot;</span>))</span>
<span id="cb158-3"><a href="#cb158-3" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L, <span class="at">modelColumn =</span> <span class="st">&quot;wrong&quot;</span>,   <span class="at">resultsColumnNamePrefix =</span> <span class="st">&quot;wrong_&quot;</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb158-4"><a href="#cb158-4" tabindex="-1"></a></span>
<span id="cb158-5"><a href="#cb158-5" tabindex="-1"></a>volcanoWrong <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb158-6"><a href="#cb158-6" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>wrong_genotypeD8,</span>
<span id="cb158-7"><a href="#cb158-7" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb158-8"><a href="#cb158-8" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb158-9"><a href="#cb158-9" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb158-10"><a href="#cb158-10" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb158-11"><a href="#cb158-11" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb158-12"><a href="#cb158-12" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;wrong analysis: &quot;</span>,<span class="fu">sum</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>wrong_genotypeD8<span class="sc">$</span>adjPval <span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>),<span class="st">&quot; DA&quot;</span>))</span>
<span id="cb158-13"><a href="#cb158-13" tabindex="-1"></a></span>
<span id="cb158-14"><a href="#cb158-14" tabindex="-1"></a>volcanoWrong</span></code></pre></div>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<p>Note, that the analysis where we ignore that we have multiple
technical repeats for each bio-repeat returns many significant DA
proteins because we act as if we have much more independent
observations.</p>
</div>
<div id="correct-analysis" class="section level3" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Correct
analysis</h3>
<ul>
<li><p>Mixed Models can model the correlation structure in the
data</p></li>
<li><p>They can acknowledge that protein expression values from runs
from the same biological repeat are more alike than protein expression
values from runs of different biological repeats.</p></li>
<li><p>Mixed models can also be used when inference between and within
blocks is needed.</p></li>
<li><p>For the francisella example the formula in msqrob2 becomes:
(formula: ~ genotype + (1|biorep)).</p></li>
</ul>
<p><span class="math display">\[ \left\{\begin{array}{rcl}
y_{ir} &amp;=&amp; \beta_0 + \beta_{wt}X_{wt,i} + b_i + \epsilon_{ir}\\
b_i &amp; \sim &amp; N(0,\sigma_b^2)\\
\epsilon_{ir} &amp;\sim&amp; N(0,\sigma_\epsilon^2)
\end{array}\right.\]</span></p>
<div class="sourceCode" id="cb159"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">formula =</span> <span class="sc">~</span>genotype <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>biorep), <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Warning: &#39;experiments&#39; dropped; see &#39;drops()&#39;</code></pre>
<div class="sourceCode" id="cb161"><pre
class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">&quot;genotypeD8 = 0&quot;</span>, <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">&quot;genotypeD8&quot;</span>))</span>
<span id="cb161-2"><a href="#cb161-2" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(<span class="at">object =</span> pe, <span class="at">i =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">contrast =</span> L, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb161-3"><a href="#cb161-3" tabindex="-1"></a></span>
<span id="cb161-4"><a href="#cb161-4" tabindex="-1"></a>volcanoMixed <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb161-5"><a href="#cb161-5" tabindex="-1"></a>    <span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>genotypeD8,</span>
<span id="cb161-6"><a href="#cb161-6" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb161-7"><a href="#cb161-7" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb161-8"><a href="#cb161-8" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">cex =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb161-9"><a href="#cb161-9" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb161-10"><a href="#cb161-10" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb161-11"><a href="#cb161-11" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;correct analysis: &quot;</span>,<span class="fu">sum</span>(<span class="fu">rowData</span>(pe[[<span class="st">&quot;protein&quot;</span>]])<span class="sc">$</span>genotypeD8<span class="sc">$</span>adjPval <span class="sc">&lt;</span><span class="fl">0.05</span>, <span class="at">na.rm=</span><span class="cn">TRUE</span>),<span class="st">&quot; DA&quot;</span>))</span>
<span id="cb161-12"><a href="#cb161-12" tabindex="-1"></a>volcanoMixed</span></code></pre></div>
<pre><code>## Warning: Removed 46 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="pda_quantification_inference_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
<ul>
<li><p>Note, that the statistical inference with mixed models is still a
bit too liberal because it:</p>
<ul>
<li>hypothesis tests are only valid asymptotically (large number of
biorepeats).</li>
<li>in small samples the between biorepeat variance is sometimes
estimated to be zero due to estimation uncertainty and for these
proteins the model still acts as if all 18 protein expression values are
independent.</li>
</ul></li>
<li><p>Mixed models are also very useful to analyse data from large
labeled experiments <span class="citation">(Vandenbulcke and Clement
2025)</span>.</p></li>
<li><p>But, they are beyond the scope of the lecture series.</p></li>
</ul>
<p>CONSULT BIOSTATISTICIAN IN CASE OF EXPERIMENTS WITH
PSEUDO-REPLICATES, TECHNICAL REPEATS, COMPLEX DESIGNS, …</p>
</div>
</div>
</div>
<div id="software-code" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Software &amp;
code</h1>
<ul>
<li><p>Our R/Bioconductor package <a
href="https://www.bioconductor.org/packages/release/bioc/html/msqrob2.html">msqrob2</a>
can be used in R markdown scripts or with GUI/shinyApps <a
href="https://github.com/statOmics/QFeaturesGUI">QFeaturesGUI</a> (+)
and <a href="https://github.com/statOmics/msqrob2gui">msqrob2gui</a> (+:
forked from the UCLouvain-CBIO lab).</p></li>
<li><p>GUIs are intended as a introduction to the key concepts of
proteomics data analysis for users who have no experience in R.</p></li>
<li><p>However, learning how to code data analyses in R markdown scripts
is key for open en reproducible science and for reporting your
proteomics data analyses and interpretation in a reproducible
way.</p></li>
<li><p>More information on our tools can be found in our papers <span
class="citation">(L. J. Goeminne, Gevaert, and Clement 2016)</span>,
<span class="citation">(L. J. E. Goeminne et al. 2020)</span>, <span
class="citation">(Sticker et al. 2020)</span> and <span
class="citation">(Vandenbulcke and Clement 2025)</span>. Please refer to
our work when using our tools.</p></li>
</ul>
<!--
- Clips on the code on importing the data and preprocessing can be found in [Part I Preprocessing](./pda_quantification_preprocessing.html)

- A clip on the code for modelling and statistical inference with msqrob2 is included below

<iframe width="560" height="315"
src="https://www.youtube.com/embed/eXxIdzGOPgY"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>
-->
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-goeminne2020" class="csl-entry">
Goeminne, L. J. E., A. Sticker, L. Martens, K. Gevaert, and L. Clement.
2020. <span>“<span
class="nocase"><span>M</span><span>S</span>q<span>R</span>ob
<span>T</span>akes the <span>M</span>issing <span>H</span>urdle:
<span>U</span>niting <span>I</span>ntensity- and
<span>C</span>ount-<span>B</span>ased
<span>P</span>roteomics</span>.”</span> <em>Anal Chem</em> 92 (9):
6278–87.
</div>
<div id="ref-goeminne2016" class="csl-entry">
Goeminne, L. J., K. Gevaert, and L. Clement. 2016. <span>“<span
class="nocase"><span>P</span>eptide-level <span>R</span>obust
<span>R</span>idge <span>R</span>egression <span>I</span>mproves
<span>E</span>stimation, <span>S</span>ensitivity, and
<span>S</span>pecificity in <span>D</span>ata-dependent
<span>Q</span>uantitative <span>L</span>abel-free <span>S</span>hotgun
<span>P</span>roteomics</span>.”</span> <em>Mol Cell Proteomics</em> 15
(2): 657–68.
</div>
<div id="ref-sticker2020" class="csl-entry">
Sticker, A., L. Goeminne, L. Martens, and L. Clement. 2020. <span>“<span
class="nocase"><span>R</span>obust <span>S</span>ummarization and
<span>I</span>nference in <span>P</span>roteome-wide
<span>L</span>abel-free <span>Q</span>uantification</span>.”</span>
<em>Mol Cell Proteomics</em> 19 (7): 1209–19.
</div>
<div id="ref-vandenbulcke2025" class="csl-entry">
Vandenbulcke, C., S. Vanderaa, and L. Clement. 2025. <span>“msqrob2TMT:
Robust Linear Mixed Models for Inferring Differential Signals in Tandem
Mass Tag-Based Proteomics.”</span> <em>Molecular &amp; Cellular
Proteomics</em> 24 (3): e10101–1. <a
href="https://doi.org/10.1016/j.mcpro.2025.00101-X">https://doi.org/10.1016/j.mcpro.2025.00101-X</a>.
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Statistical Methods for Quantitative MS-based Proteomics: Part II. Differential Abundance Analysis"
author: "Lieven Clement"
date: "[statOmics](https://statomics.github.io), Ghent University"
output:
    html_document:
      code_download: true
      theme: flatly
      toc: true
      toc_float: true
      highlight: tango
      number_sections: true
    pdf_document:
      toc: true
      number_sections: true
linkcolor: blue
urlcolor: blue
citecolor: blue

bibliography: msqrob2.bib

---

<a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-nc-sa/4.0/88x31.png" /></a>

This is part of the online course [Proteomics Data Analysis (PDA)](https://statomics.github.io/PDA/)

<iframe width="560" height="315"
src="https://www.youtube.com/embed/1mhg3BCuEm8"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

- [Playlist PDA Preprocessing](https://www.youtube.com/watch?v=1mhg3BCuEm8&list=PLZH1hP8_LbJKqnPSS4hxTkn-tQolSCGjP)

# Outline {-}

- Francisella tularensis Example
- Hypothesis testing
- Multiple testing
- Moderated statistics
- Experimental design


Note, that the R-code is included for learners who are aiming to develop R/markdown scripts to automate their quantitative proteomics data analyses.
According to the target audience of the course we either work with a graphical user interface (GUI) in a R/shiny App msqrob2gui (e.g. Proteomics Bioinformatics course of the EBI and the Proteomics Data Analysis course at the Gulbenkian institute) or with R/markdowns scripts (e.g. Bioinformatics Summer School at UCLouvain or the Statistical Genomics Course at Ghent University). 

---

# Francisella tularensis experiment

<iframe width="560" height="315"
src="https://www.youtube.com/embed/TW5rk1y7aOc"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

```{r echo=FALSE,out.width="50%"}
knitr::include_graphics("./figures/francisella.jpg")
```

```{r echo=FALSE}
knitr::include_graphics("./figures/tularemia_lesion.jpg")
```

- Pathogen: causes tularemia
- Metabolic adaptation key for intracellular life cycle of pathogenic microorganisms. 
- Upon entry into host cells quick phasomal escape and active multiplication in cytosolic compartment.
- Franciscella is auxotroph for several amino acids, including arginine. 
- Inactivation of arginine transporter delayed bacterial phagosomal escape and intracellular multiplication. 
- Experiment to assess difference in proteome using 3 WT vs 3 ArgP KO mutants


## Import the data in R 

<iframe width="560" height="315"
src="https://www.youtube.com/embed/2Ljd95IE_AY"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>


<details><summary> Click to see code </summary><p>
1. Load libraries 

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(limma)
library(QFeatures)
library(msqrob2)
library(plotly)
library(ggplot2)
library(data.table)
```

2. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesTable <- fread("https://raw.githubusercontent.com/statOmics/PDA/data/quantification/francisella/peptides.txt")
int64 <- which(sapply(peptidesTable,class) == "integer64")
for (j in int64) peptidesTable[[j]] <- as.numeric(peptidesTable[[j]])                      
```

3. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
quantCols <- grep("Intensity ", names(peptidesTable))
```

4. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  assayData = peptidesTable,
  fnames = 1,
  quantCols =  quantCols,
  name = "peptideRaw")
rm(peptidesTable)
gc()
gc()
```

5. Update data with information on design

```{r}
colData(pe)$genotype <- pe[[1]] |> 
  colnames() |> 
  substr(12,13) |>
  as.factor() |> 
  relevel("WT")

pe |> colData()
```

</p></details>

## Preprocessing

<details><summary> Click to see code to log-transfrom the data </summary><p>

1. Log transform

  - Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

  - Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

2. Filtering

  - We remove peptides that could not be mapped to a protein or that map to multiple proteins (the protein identifier contains multiple identifiers separated by a `;`).

```{r}
pe <- filterFeatures(
    pe, ~ Proteins != "" & ## Remove failed protein inference
        !grepl(";", Proteins)) ## Remove protein groups
```

  - Remove reverse sequences (decoys) and contaminants. Note that this is indicated by the column names Reverse and depending on the version of maxQuant with Potential.contaminants or Contaminants.


```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Contaminant != "+")
```

  - Drop peptides that were identified in less than three sample. We tolerate the following proportion of NAs: pNA = (n-3)/n. 

```{r}
nObs <- 3
n <- ncol(pe[["peptideLog"]])
pNA <- (n-nObs)/n
pe <- filterNA(pe, pNA = pNA, i = "peptideLog")
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.

3. Normalization by median centering

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```


4. Summarization. We use the standard sumarisation in aggregateFeatures, which is a
robust summarisation method.

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
    i = "peptideNorm", 
    fcol = "Proteins", 
    na.rm = TRUE,
    name = "protein")
```

5. Filter proteins that contain many missing values 

We want to have at least 4 observed proteins so that most proteins have at least 2 observations in each group.  
So we tolerate a proportion of (n-4)/n NAs. 


```{r}
nObs <- 4
n <- ncol(pe[["protein"]])
pNA <- (n-nObs)/n
pe <- filterNA(pe, pNA = pNA, i = "protein")
```

Plot of preprocessed data 

```{r}
pe[["peptideNorm"]] |> 
  assay() |>
  as.data.frame() |>
  gather(sample, intensity) |> 
  mutate(genotype = colData(pe)[sample,"genotype"]) |>
  ggplot(aes(x = intensity,group = sample,color = genotype)) + 
    geom_density() +
    ggtitle("Peptide-level")

pe[["protein"]] |> 
  assay() |>
  as.data.frame() |>
  gather(sample, intensity) |> 
  mutate(genotype = colData(pe)[sample,"genotype"]) |>
  ggplot(aes(x = intensity,group = sample,color = genotype)) + 
    geom_density() +
    ggtitle("Protein-level")
```
</p></details>

## Summarized data structure

### Design

```{r}
pe |> 
  colData() |> 
  knitr::kable()
```

- WT vs KO 
- 3 vs 3 repeats 

### Summarized intensity matrix

```{r}
pe[["protein"]] |> 
  assay() |> 
  head() |> 
  knitr::kable()
```

- `r nrow(pe[["protein"]])` proteins 

### Hypothesis testing: a single protein 

<iframe width="560" height="315"
src="https://www.youtube.com/embed/myP6SUlSwsM"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

```{r echo=FALSE}
if ("pi"%in%ls()) rm("pi")
kopvoeter<-function(x,y,angle=0,l=.2,cex.dot=.5,pch=19,col="black")
{
angle=angle/180*pi
points(x,y,cex=cex.dot,pch=pch,col=col)
lines(c(x,x+l*cos(-pi/2+angle)),c(y,y+l*sin(-pi/2+angle)),col=col)
lines(c(x+l/2*cos(-pi/2+angle),x+l/2*cos(-pi/2+angle)+l/4*cos(angle)),c(y+l/2*sin(-pi/2+angle),y+l/2*sin(-pi/2+angle)+l/4*sin(angle)),col=col)
lines(c(x+l/2*cos(-pi/2+angle),x+l/2*cos(-pi/2+angle)+l/4*cos(pi+angle)),c(y+l/2*sin(-pi/2+angle),y+l/2*sin(-pi/2+angle)+l/4*sin(pi+angle)),col=col)
lines(c(x+l*cos(-pi/2+angle),x+l*cos(-pi/2+angle)+l/2*cos(-pi/2+pi/4+angle)),c(y+l*sin(-pi/2+angle),y+l*sin(-pi/2+angle)+l/2*sin(-pi/2+pi/4+angle)),col=col)
lines(c(x+l*cos(-pi/2+angle),x+l*cos(-pi/2+angle)+l/2*cos(-pi/2-pi/4+angle)),c(y+l*sin(-pi/2+angle),y+l*sin(-pi/2+angle)+l/2*sin(-pi/2-pi/4+angle)),col=col)
}

par(mar=c(0,0,0,0),mai=c(0,0,0,0))
plot(0,0,xlab="",ylab="",xlim=c(0,10),ylim=c(0,10),col=0,xaxt="none",yaxt="none",axes=FALSE)
rect(0,6,10,10,border="red",lwd=2)
text(.5,8,"population",srt=90,col="red",cex=2)
symbols (3, 8, circles=1.2, col="red",add=TRUE,fg="red",inches=FALSE,lwd=2)
set.seed(330)
grid=seq(0,1,.01)

for (i in 1:50)
{
	angle1=runif(n=1,min=0,max=360)
	angle2=runif(n=1,min=0,max=360)
	radius=sample(grid,prob=grid^2*pi/sum(grid^2*pi),size=1)
	kopvoeter(3+radius*cos(angle1/180*pi),8+radius*sin(angle1/180*pi),angle=angle2)
}
text(7.5,8,"Effect of arginine def. in population",col="red",cex=1.2)

rect(0,0,10,4,border="blue",lwd=2)
text(.5,2,"sample",srt=90,col="blue",cex=2)
symbols (3, 2, circles=1.2, col="red",add=TRUE,fg="blue",inches=FALSE,lwd=2)
for (i in 0:1)
	for (j in 0:2)
{

	kopvoeter(2.5+j*(3.9-2.1)/4,1.5+i)
}
text(7.5,2,"Effect of arginine def. in sample",col="blue",cex=1.2)

arrows(3,5.9,3,4.1,col="black",lwd=3)
arrows(7,4.1,7,5.9,col="black",lwd=3)
text(1.5,5,"Exp. Design",col="black",cex=1.2)
text(8.5,5,"Estimation \n Inference ",col="black",cex=1.2)
text(7.5,.5,"Data exploration",col="black",cex=1.2)
```

#### T-test

<iframe width="560" height="315"
src="https://www.youtube.com/embed/cVw5kdSRZCE"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>


$$
 \log_2 \text{FC} = \bar{y}_{p1}-\bar{y}_{p2}
$$

$$
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
$$

$$
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
$$

If we can assume equal variance in both treatment groups:

$$
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
$$

```{r}
WP_003023392 <- data.frame(
    intensity = assay(pe[["protein"]]["WP_003023392",]) |> 
      c(), 
    genotype = colData(pe)[,1]) 

WP_003023392 |> 
  ggplot(aes(x=genotype,y=intensity)) + 
  geom_point() +
  ggtitle("Protein WP_003023392")
```

```{r echo=FALSE}
lmHlp <- lm(intensity ~ genotype, data = WP_003023392)
```

$$
t=\frac{\log_2\widehat{\text{FC}}}{\text{se}_{\log_2\widehat{\text{FC}}}}=\frac{`r format(summary(lmHlp)$coef[2,1], digit = 3)`}{`r format(summary(lmHlp)$coef[2,2], digit = 3)`}=`r format(summary(lmHlp)$coef[2,3], digit = 3)`
$$

- Is t = `r format(summary(lmHlp)$coef[2,3], digit = 3)` indicating that
there is an effect?

- How likely is it to observe
t = `r format(summary(lmHlp)$coef[2,3], digit = 3)` when there is no effect of the argP KO on the protein expression?

#### Null hypothesis ($H_0$) and alternative hypothesis ($H_1$)

- With data we can never prove a hypothesis (falsification principle of Popper)
- With data we can only reject a hypothesis 

- In general we start from *alternative hypothese* $H_1$: we want to show an effect of the KO on a protein

<center>
$H_1$: On average the protein abundance in WT is different from that in KO
</center>

- But, we will assess this by falsifying the opposite: 
<center>
$H_0$: On average the protein abundance in WT is equal to that in KO<-
</center>


```{r}
t.test(intensity ~ genotype, data = WP_003023392, var.equal=TRUE)
```

- How likely is it to observe an equal or more extreme effect than the one observed in the sample when the null hypothesis is true?
- When we make assumptions about the distribution of our test statistic we can quantify this probability: *p-value*. 
The p-value will only be calculated correctly if the underlying assumptions hold!
- When we repeat the experiment, the probability to observe a fold change for this gene that is more extreme than a `r format(2^abs(lmHlp$coef[2]),digits=3)` fold ($\log_2 FC=`r format(lmHlp$coef[2],digits=3)`$) down or up regulation by random change (if $H_0$ is true) is `r round(summary(lmHlp)$coef[2,4]*1e6,0)` out of 1 000 000.  
- If the p-value is below a significance threshold $\alpha$ we reject the null hypothesis. *We control the probability on a false positive result at the $\alpha$-level (type I error)*

- Note, that the p-values are uniform under the null hypothesis, i.e. when $H_0$ is true all p-values are equally likely. 

## Multiple hypothesis testing

<iframe width="560" height="315"
src="https://www.youtube.com/embed/cLn-CFyA6ps"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

- Consider testing DA for all $m=1066$ proteins simultaneously
- What if we assess each individual test at level $\alpha$?
$\rightarrow$ Probability to have a false positive (FP) among all m simultatenous
test $>>>  \alpha= 0.05$

- Indeed for each non DA protein we have a probability of 5% to return a FP.
- In a typical experiment the majority of the proteins are non DA. 
- So an upperbound of the expected FP is $m \times \alpha$ or $1066 \times 0.05=`r round(1066*0.05,0)`$. 

$\rightarrow$ Hence, we are bound to call many false positive proteins each time we run the experiment.

### Multiple testing

#### Family-wise error rate

<iframe width="560" height="315"
src="https://www.youtube.com/embed/IL_eUSyRDRA"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

The family-wise error rate (FWER) addresses the multiple testing issue by no longer controlling the individual type I error for each protein, instead it controls:  

\[
   \text{FWER} = 
   \text{P}\left[FP \geq 1 \right].
\]

The Bonferroni method is widely used to control the type I error: 

- assess each test at 
\[\alpha_\text{adj}=\frac{\alpha}{m}\]
- or use adjusted p-values and compare them to $\alpha$: 
\[p_\text{adj}=\text{min}\left(p \times m,1\right)\]

Problem, the method is very conservative! 

#### False discovery rate

<iframe width="560" height="315"
src="https://www.youtube.com/embed/envDqvEwRcc"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

- FDR: Expected proportion of false positives on the total number of positives you return.
- An FDR of 1% means that on average we expect 1% false positive proteins in the list of proteins that are called significant.
- Defined by Benjamini and Hochberg in their seminal paper Benjamini, Y. and Hochberg, Y. (1995). "Controlling the false discovery rate: a practical and powerful approach to multiple testing". Journal of the Royal Statistical Society Series B, 57 (1): 289–300. 

The **False Discovery Proportion (FDP)** is the fraction of false positives that are returned, i.e. 

\[
FDP = \frac{FP}{R}
\]

- However, this quantity cannot be observed because in practice we only know the number of proteins for which we rejected $H_0$, $R$. 
- But, we do not know the number of false positives, $FP$.

Therefore, Benjamini and Hochberg, 1995, defined The **False Discovery Rate (FDR)** as
\[
   \text{FDR} = \text{E}\left[\frac{FP}{R}\right] =\text{E}\left[\text{FDP}\right]
\]
the expected FDP. 

- Controlling the FDR allows for more discoveries (i.e. longer lists with significant results), while the fraction of false discoveries among the significant results in well controlled on average. As a consequence, more of the true positive hypotheses will be detected.


#### Intuition of BH-FDR procedure

Consider $m = 1000$ tests

- Suppose that a researcher rejects all null hypotheses for which $p < 0.01$. 

- If we use $p < 0.01$, we expect $0.01 \times m_0$ tests to return false positives. 
- A conservative estimate of the number of false positives that we can expect can be obtained by considering that the null hypotheses are true for all features, $m_0 = m =  1000$. 
- We then would expect $0.01 \times 1000 = 10$ false positives ($FP=10$).

- Suppose that the researcher found 200 genes with $p<0.01$ ($R=200$).

- The proportion of false positive results (FDP = false positive proportion) among the list of $R=200$ genes can then be estimated as
 \[
   \widehat{\text{FDP}}=\frac{FP}{R}=\frac{10}{200}=\frac{0.01 \times 1000}{200} = 0.05.
 \]


#### Benjamini and Hochberg (1995) procedure for controlling the FDR at $\alpha$

1. Let $p_{(1)}\leq \ldots \leq p_{(m)}$ denote the ordered $p$-values.

2. Find the largest integer $k$ so that 
$$
\frac{p_{(k)} \times m}{k} \leq \alpha
$$
$$\text{or}$$
$$
p_{(k)} \leq k \times \alpha/m
$$

3. If such a $k$ exists, reject the $k$ null hypotheses associated with $p_{(1)}, \ldots, p_{(k)}$.
Otherwise none of the null hypotheses is rejected.

The adjusted $p$-value (also known as the $q$-value in FDR literature):
 $$
   q_{(i)}=\tilde{p}_{(i)} = \min\left[\min_{j=i,\ldots, m}\left(m p_{(j)}/j\right), 1 \right].
 $$
 In the hypothetical example above: $k=200$, $p_{(k)}=0.01$, $m=1000$ and $\alpha=0.05$.

#### Francisella Example

<iframe width="560" height="315"
src="https://www.youtube.com/embed/B3BmnOLkYg4"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

<details><summary> Click to see code </summary><p>
```{r}
ttestMx <- function(y,group) {
    test <- try(t.test(y[group],y[!group],var.equal=TRUE),silent=TRUE)
    if(is(test,"try-error")) {
      return(c(log2FC=NA,se=NA,tstat=NA,p=NA))
      } else {
      return(c(log2FC= (test$estimate%*%c(1,-1)),se=test$stderr,tstat=test$statistic,pval=test$p.value))
      }
 }
 
 res <- apply(
    assay(pe[["protein"]]), 
    1, 
    ttestMx,
    group = colData(pe)$genotype=="D8") |> 
  t() 
 
 colnames(res) <- c("logFC","se","tstat","pval")
 
 res <- res |> 
   as.data.frame() |> 
   na.exclude() |> 
   arrange(pval)
 
 res$adjPval <- p.adjust(res$pval, "fdr")
 
 alpha <- 0.05

 res$adjAlphaForm <- paste0(1:nrow(res)," x ",alpha,"/",nrow(res))

 res$adjAlpha <- alpha * (1:nrow(res))/nrow(res) 

 res$"pval < adjAlpha" <- res$pval < res$adjAlpha 

 res$"adjPval < alpha" <- res$adjPval < alpha 
```
</p></details>

FWER: Bonferroni method:$\alpha_\text{adj} = \alpha/m = 0.05 / `r nrow(res)`= `r round(alpha/nrow(res),5)`$

```{r echo=FALSE} 
head(res[,-c(2:3)],sum(res$adjPval < alpha)+2) |> 
  knitr::kable()
```
| ... | ... | ... | ... | ... | ... | ... | ... |
|WP_003040562 | 0.0039480| 0.9976429| 0.9985797|1065 x 0.05/1066 | 0.0499531|FALSE           |FALSE    
|WP_003041130 | 0.0002941| 0.9992812| 0.9992812|1066 x 0.05/1066 |     0.05|FALSE           |FALSE           |




#### Results
<details><summary> Click to see code </summary><p>
```{r}
volcanoT <- res |> 
  ggplot(aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() 
```
</p></details>


```{r}
volcanoT
```

## Moderated Statistics

<iframe width="560" height="315"
src="https://www.youtube.com/embed/_Q11LXDy0xU"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

Problems with ordinary t-test

<details><summary> Click to see code </summary><p>
```{r}
problemPlots <- list() 
problemPlots[[1]] <- res |> 
  ggplot(aes(x = logFC, y = se, color = adjPval < 0.05)) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() 

for (i in 2:3)
{
problemPlots[[i]] <- colData(pe) |> 
    as.data.frame() |> 
    mutate(intensity = pe[["protein"]][rownames(res)[i],] |> 
             assay() |> 
             c()
           ) |> 
    ggplot(aes(x=genotype,y=intensity)) +
    geom_point() + 
    ylim(-3,0) +
    ggtitle(rownames(res)[i])
}
```
</p></details>

```{r}
problemPlots
```

A general class of moderated test statistics is given by
 \[
   T_g^{mod} = \frac{\bar{Y}_{g1} - \bar{Y}_{g2}}{C \quad \tilde{S}_g} ,
 \]
 where $\tilde{S}_g$ is a moderated standard deviation estimate. 

- $C$ is a constant depending on the design e.g. $\sqrt{1/{n_1}+1/n_2}$ for a t-test and of another form for linear models.
- $\tilde{S}_g=S_g+S_0$: add small positive constant to denominator of t-statistic. 
- This can be adopted in Perseus. 

<details><summary> Click to see code </summary><p>
```{r}
simI<-sapply(res$se/sqrt(1/3+1/3),function(n,mean,sd) rnorm(n,mean,sd),n=6,mean=0) |> 
  t()

resSim <- apply(
    simI, 
    1, 
    ttestMx,
    group = colData(pe)$genotype=="D8") |> 
  t() 
 colnames(resSim) <- c("logFC","se","tstat","pval")
 resSim <- as.data.frame(resSim)
 tstatSimPlot <- resSim |> 
   ggplot(aes(x=tstat)) +
     geom_histogram(aes(y=..density.., fill=..count..),bins=30) +
     stat_function(fun=dt,
    color="red",
    args=list(df=4)) + 
   ylim(0,.6) +
   ggtitle("t-statistic")

 
 resSim$C <- sqrt(1/3+1/3) 
 resSim$sd <- resSim$se/resSim$C 
 tstatSimPerseus <- resSim |> 
   ggplot(aes(x=logFC/((sd+.1)*C))) +
     geom_histogram(aes(y=..density.., fill=..count..),bins=30) +
     stat_function(fun=dt,
                   color="red",
                  args=list(df=4)) + 
     ylim(0,.6) +
    ggtitle("Perseus")
```

</p></details>

```{r}
gridExtra::grid.arrange(tstatSimPlot,tstatSimPerseus,nrow=1)
```

- The choice of $S_0$ in Perseus is ad hoc and the t-statistic is no-longer t-distributed. 
- Permutation test, but is difficult for more complex designs.
- Allows for Data Dredging because user can choose $S_0$ 


### Empirical Bayes 


```{r echo=FALSE, out.width="50%"}
knitr::include_graphics("./figures/limmaShrinkage.png")
```

Figure courtesy to Rafael Irizarry

$$
   T_g^{mod} = \frac{\bar{Y}_{g1} - \bar{Y}_{g2}}{C \quad \tilde{S}_g} ,
 $$

- **empirical Bayes** theory provides formal framework for borrowing strength across proteins,
- Implemented in popular bioconductor package **limma** and **msqrob2**

$$
  \tilde{S}_g=\sqrt{\frac{d_gS_g^2+d_0S_0^2}{d_g+d_0}},
$$

- $S_0^2$: common variance (over all proteins) 
- Moderated t-statistic is t-distributed with $d_0+d_g$ degrees of freedom. 
- Note that the degrees of freedom increase by borrowing strength across proteins!

<details><summary> Click to see the code </summary><p>   

1. We model the protein level expression values using the `msqrob` function.
By default `msqrob2` estimates the model parameters using robust regression.

We will model the data with a different group mean for every genotype. 
The group is incoded in the variable `genotype` of the colData. 
We can specify this model by using a formula with the factor `genotype` as its predictor: 
`formula = ~genotype`.

Note, that a formula always starts with a symbol '~'.
```{r warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~genotype)
```

2. Inference 

We first explore the design of the model that we specified using the the package `ExploreModelMatrix` 

```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~genotype)$plotlist
```

We have two model parameters, the (Intercept) and genotypeD8. 
This results in a model with two group means: 

1. For the wild type (WT) the expected value (mean) of the log2 transformed intensity y for a protein will be modelled using 

$$\text{E}[Y\vert \text{genotype}=\text{WT}] = \text{(Intercept)}$$ 

2. For the knockout genotype D8 the expected value (mean) of the log2 transformed intensity y for a protein will be modelled using 

$$\text{E}[Y\vert \text{genotype}=\text{D8}] = \text{(Intercept)} + \text{genotypeD8}$$ 

The average log2FC between D8 and WT is thus
$$\log_2\text{FC}_{D8-WT}= \text{E}[Y\vert \text{genotype}=\text{D8}] - \text{E}[Y\vert \text{genotype}=\text{WT}] = \text{genotypeD8}
$$

Hence, assessing the null hypothesis that there is no differential abundance between D8 and WT can be reformulated as

$$H_0:  \text{genotypeD8}=0$$
We can implement a hypothesis test for each protein in msqrob2 using the code below: 

```{r}
L <- makeContrast("genotypeD8 = 0", parameterNames = c("genotypeD8"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
```

We can show the list with all significant DE proteins at the 5% FDR using 
```{r}
rowData(pe[["protein"]])$genotypeD8 |> 
  arrange(pval) |>
  filter(adjPval<0.05)
```

We can also visualise the results using a volcanoplot

```{r}
volcano <- ggplot(
    rowData(pe[["protein"]])$genotypeD8,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle("msqrob2")
```
</p></details>

```{r}
gridExtra::grid.arrange(
  volcanoT +    
    xlim(-3,3) +
  ggtitle("ordinary t-test"),
  volcano +     
    xlim(-3,3)
,nrow=2)
```

- The volcano plot opens up when using the EB variance estimator

-  Borrowing strength to estimate the variance using empirical Bayes solves the issue of returning proteins with a low fold change as significant due to a low variance. 


### Shrinkage of the  variance and moderated t-statistics

```{r}
qplot(
  sapply(rowData(pe[["protein"]])$msqrobModels,getSigma),
  sapply(rowData(pe[["protein"]])$msqrobModels,getSigmaPosterior)) +
  xlab("SD") +
  ylab("moderated SD") +
  geom_abline(intercept = 0,slope = 1) +
  geom_hline(yintercept = ) 
```

- Small variances are shrunken towards the common variance resulting in large EB variance estimates
- Large variances are shrunken towards the common variance resulting in smaller EB variance estimates 
- Pooled degrees of freedom of the EB variance estimator are larger because information is borrowed across proteins to estimate the variance

## Plots 

```{r}
sigNames <- rowData(pe[["protein"]])$genotypeD8 |>
    rownames_to_column("protein") |>
    filter(adjPval < 0.05) |>
    pull(protein)
heatmap(assay(pe[["protein"]])[sigNames, ])
```


```{r, warning=FALSE, message=FALSE}
for (protName in sigNames)
    {
        pePlot <- pe[protName, , c("peptideNorm", "protein")]
        pePlotDf <- data.frame(longForm(pePlot))
        pePlotDf$assay <- factor(pePlotDf$assay,
            levels = c("peptideNorm", "protein")
        )
        pePlotDf$genotype <- as.factor(colData(pePlot)[pePlotDf$colname, "genotype"])

        # plotting
        p1 <- ggplot(
            data = pePlotDf,
            aes(x = colname, y = value, group = rowname)
        ) +
            geom_line() +
            geom_point() +
            facet_grid(~assay) +
            theme(axis.text.x = element_text(angle = 70, hjust = 1, vjust = 0.5)) +
            ggtitle(protName)
        print(p1)

        # plotting 2
        p2 <- ggplot(pePlotDf, aes(x = colname, y = value, fill = genotype)) +
            geom_boxplot(outlier.shape = NA) +
            geom_point(
                position = position_jitter(width = .1),
                aes(shape = rowname)
            ) +
            scale_shape_manual(values = 1:nrow(pePlotDf)) +
            labs(title = protName, x = "sample", y = "peptide intensity (log2)") +
            theme(axis.text.x = element_text(angle = 70, hjust = 1, vjust = 0.5)) +
            facet_grid(~assay)
        print(p2)
}
```

# Experimental Design

## Sample size 

<iframe width="560" height="315"
src="https://www.youtube.com/embed/uELgkzDjVRY"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>

$$
 \log_2 \text{FC} = \bar{y}_{p1}-\bar{y}_{p2}
$$

$$
T_g=\frac{\log_2 \text{FC}}{\text{se}_{\log_2 \text{FC}}}
$$

$$
T_g=\frac{\widehat{\text{signal}}}{\widehat{\text{Noise}}}
$$

If we can assume equal variance in both treatment groups:

$$
\text{se}_{\log_2 \text{FC}}=\text{SD}\sqrt{\frac{1}{n_1}+\frac{1}{n_2}}
$$

$\rightarrow$ if number of bio-repeats increases we have a higher power!

- cfr. Study of tamoxifen treated Estrogen Recepter (ER) positive breast cancer patients

## Randomized complete block designs

<iframe width="560" height="315"
src="https://www.youtube.com/embed/HE4KWfIOVp0"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>


\[\sigma^2= \sigma^2_{bio}+\sigma^2_\text{lab} +\sigma^2_\text{extraction} + \sigma^2_\text{run} + \ldots\]

- Biological: fluctuations in protein level between mice, fluctations in protein level between cells, ...
- Technical: cage effect, lab effect, week effect, plasma extraction, MS-run, ...

### Nature methods: Points of significance - Blocking 

[https://www.nature.com/articles/nmeth.3005.pdf](https://www.nature.com/articles/nmeth.3005.pdf)


### Mouse example 


```{r echo=FALSE, out.width="50%"}
knitr::include_graphics("./figures/mouseTcell_RCB_design.png")
```

Duguet et al. (2017) MCP 16(8):1416-1432. doi: 10.1074/mcp.m116.062745

- All treatments of interest are present within block!
- We can estimate the effect of the treatment within block!

To illustrate the power of blocking we have subsetted the data of Duguet et al. in a 

- completely randomized design with 
    - four mice for which we only have measurements on the ordinary T-cells 
    - four mice for which we only have measurements on the regulatory T-cells 
    
- randomized complete block design with four mice for which we both have
    
    - measurements on ordinary T-cells as well as
    - measurements on regulatory T-cells




### Data 
<details><summary> Click to see code  </summary><p>
```{r, warning=FALSE, message=FALSE}
peptidesTable <- fread("https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesRCB.txt")
int64 <- which(sapply(peptidesTable,class) == "integer64")
for (j in int64) peptidesTable[[j]] <- as.numeric(peptidesTable[[j]]) 

peptidesTable2 <- fread("https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptidesCRD.txt")
int64 <- which(sapply(peptidesTable2,class) == "integer64")
for (j in int64) peptidesTable2[[j]] <- as.numeric(peptidesTable2[[j]]) 

peptidesTable3 <- fread("https://raw.githubusercontent.com/statOmics/PDA21/data/quantification/mouseTcell/peptides.txt")
int64 <- which(sapply(peptidesTable3,class) == "integer64")
for (j in int64) peptidesTable3[[j]] <- as.numeric(peptidesTable3[[j]]) 

quantCols <- grep("Intensity ", names(peptidesTable))
pe <- readQFeatures(
  assayData = peptidesTable,
  fnames = 1,
  quantCols =  quantCols,
  name = "peptideRaw")
rm(peptidesTable)
gc()
gc()


quantCols2 <- grep("Intensity ", names(peptidesTable2))
pe2 <- readQFeatures(
  assayData = peptidesTable2,
  fnames = 1,
  quantCols =  quantCols2,
  name = "peptideRaw")
rm(peptidesTable2)
gc()
gc()

quantCols3 <- grep("Intensity ", names(peptidesTable3))
pe3 <- readQFeatures(
  assayData = peptidesTable3,
  fnames = 1,
  quantCols =  quantCols3,
  name = "peptideRaw")
rm(peptidesTable3)
gc()
gc()

### Design
colData(pe)$celltype <- substr(
  colnames(pe[["peptideRaw"]]),
  11,
  14) |>
  unlist() |>  
  as.factor()

colData(pe)$mouse <- pe[[1]] |>
  colnames() |>
  strsplit(split="[ ]")  |>
  sapply(function(x) x[3]) |>
  as.factor()

colData(pe2)$celltype <- substr(
  colnames(pe2[["peptideRaw"]]),
  11,
  14) |>
  unlist() |>  
  as.factor()

colData(pe2)$mouse <- pe2[[1]] |>
  colnames() |>
  strsplit(split="[ ]")  |>
  sapply(function(x) x[3]) |>
  as.factor()

colData(pe3)$celltype <- substr(
  colnames(pe3[["peptideRaw"]]),
  11,
  14) |>
  unlist() |>  
  as.factor()

colData(pe3)$mouse <- pe3[[1]] |>
  colnames() |>
  strsplit(split="[ ]")  |>
  sapply(function(x) x[3]) |>
  as.factor()
```
</p></design>

### Preprocessing 


#### Log-transform

<details><summary> Click to see code to log-transfrom the data </summary><p>

- Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA

pe2 <- zeroIsNA(pe2, "peptideRaw") # convert 0 to NA

pe3 <- zeroIsNA(pe3, "peptideRaw") # convert 0 to NA
```

- Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")

pe2 <- logTransform(pe2, base = 2, i = "peptideRaw", name = "peptideLog")

pe3 <- logTransform(pe3, base = 2, i = "peptideRaw", name = "peptideLog")
```
</p></details>


#### Filtering
<details><summary> Click to see details on filtering </summary><p>

1. Remove peptides that map to multiple proteins

We remove PSMs that could not be mapped to a protein or that map
to multiple proteins (the protein identifier contains multiple
identifiers separated by a `;`).

```{r}
pe <- filterFeatures(
    pe, ~ Proteins != "" & ## Remove failed protein inference
        !grepl(";", Proteins)) ## Remove protein groups

pe2 <- filterFeatures(
    pe2, ~ Proteins != "" & ## Remove failed protein inference
        !grepl(";", Proteins)) ## Remove protein groups

pe3 <- filterFeatures(
    pe3, ~ Proteins != "" & ## Remove failed protein inference
        !grepl(";", Proteins)) ## Remove protein groups
```
2. Remove reverse sequences (decoys) and contaminants

We now remove the contaminants, peptides that map to decoy sequences, and proteins
which were only identified by peptides with modifications.

```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Potential.contaminant != "+")

pe2 <- filterFeatures(pe2,~Reverse != "+")
pe2 <- filterFeatures(pe2,~ Potential.contaminant != "+")

pe3 <- filterFeatures(pe3,~Reverse != "+")
pe3 <- filterFeatures(pe3,~ Potential.contaminant != "+")
```


3. Drop peptides that were identified in less than three sample

We keep peptides that were observed at least three times.

```{r}
nObs <- 3
n <- ncol(pe[["peptideLog"]])
pNA <- (n-nObs)/n
pe <- filterNA(pe, pNA = pNA, i = "peptideLog")
nrow(pe[["peptideLog"]])

n <- ncol(pe2[["peptideLog"]])
pNA <- (n-nObs)/n
pe2 <- filterNA(pe2, pNA = pNA, i = "peptideLog")
nrow(pe2[["peptideLog"]])

n <- ncol(pe3[["peptideLog"]])
pNA <- (n-nObs)/n
pe3 <- filterNA(pe3, pNA = pNA, i = "peptideLog")
nrow(pe3[["peptideLog"]])
```

</p></details>

#### Normalization 

<details><summary> Click to see code to normalize the data </summary><p>
```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")

pe2 <- normalize(pe2, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")


pe3 <- normalize(pe3, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```

</p></details>

#### Summarization

<details><summary> Click to see code to summarize the data </summary><p>

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")


pe2 <- aggregateFeatures(pe2,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")

pe3 <- aggregateFeatures(pe3,
 i = "peptideNorm",
 fcol = "Proteins",
 na.rm = TRUE,
 name = "protein")
```

#### Filtering proteins with too many missing values 

We want to have at least two observed protein intensities for each group so we set the minimum number of observed values at 4. 
We still have to check for the observed proteins if that is the case. 

For block design more clever filtering can be used. E.g. we could imply that we have both cell types in at least two animals... 


```{r}
nObs <- 4
n <- ncol(pe[["protein"]])
pNA <- (n-nObs)/n
pe<- filterNA(pe, pNA = pNA, i = "protein")

n <- ncol(pe2[["protein"]])
pNA <- (n-nObs)/n
pe2 <- filterNA(pe2, pNA = pNA, i = "protein")

n <- ncol(pe3[["protein"]])
pNA <- (n-nObs)/n
pe3 <- filterNA(pe3, pNA = pNA, i = "protein")
```

</p></details>

### Data Exploration: what is impact of blocking? 

<details><summary> Click to see code </summary><p>
```{r}
levels(colData(pe3)$mouse) <- paste0("m",1:7)
mdsObj3 <- plotMDS(assay(pe3[["protein"]]), plot = FALSE)
mdsOrig <- colData(pe3) |>
  as.data.frame() |>
  mutate(mds1 = mdsObj3$x,
         mds2 = mdsObj3$y,
         lab = paste(mouse,celltype,sep="_")) |>
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj3$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj3$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj3$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj3$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Original (RCB)")

levels(colData(pe)$mouse) <- paste0("m",1:4)
mdsObj <- plotMDS(assay(pe[["protein"]]), plot = FALSE)
mdsRCB <- colData(pe) |>
  as.data.frame() |>
  mutate(mds1 = mdsObj$x,
         mds2 = mdsObj$y,
         lab = paste(mouse,celltype,sep="_")) |>
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  geom_line(color = "black", linetype = "dashed") +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Randomized Complete Block (RCB)")


levels(colData(pe2)$mouse) <- paste0("m",1:8)
mdsObj2 <- plotMDS(assay(pe2[["protein"]]), plot = FALSE)
mdsCRD <- colData(pe2) |>
  as.data.frame() |>
  mutate(mds1 = mdsObj2$x,
         mds2 = mdsObj2$y,
         lab = paste(mouse,celltype,sep="_")) |>
  ggplot(aes(x = mds1, y = mds2, label = lab, color = celltype, group = mouse)) +
  geom_text(show.legend = FALSE) +
  geom_point(shape = 21) +
  xlab(
    paste0(
      mdsObj$axislabel,
      " ",
      1, 
      " (",
      paste0(
        round(mdsObj2$var.explained[1] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ylab(
    paste0(
      mdsObj$axislabel,
      " ",
      2, 
      " (",
      paste0(
        round(mdsObj2$var.explained[2] *100,0),
        "%"
        ),
      ")"
      )
    ) +
  ggtitle("Completely Randomized Design (CRD)")
```
</p></details>
```{r}
mdsOrig
mdsRCB
mdsCRD
```

- We observe that the leading fold change is according to mouse
- In the second dimension we see a separation according to cell-type 
- With the Randomized Complete Block design (RCB) we can remove the mouse effect from the analysis!




- We can isolate the between block variability from the analysis using linear model:
  
  - Formula in R
$$ 
y \sim \text{celltype} + \text{mouse}
$$
  
  - Formula 

$$
y_i = \beta_0 + \beta_\text{Treg} x_{i,\text{Treg}} + \beta_{m2}x_{i,m2} + \beta_{m3}x_{i,m3} + \beta_{m4}x_{i,m4}  + \epsilon_i
$$

with

- $x_{i,Treg}=\begin{cases}
1& \text{Treg}\\
0& \text{Tcon}
\end{cases}$

- $x_{i,m2}=\begin{cases}
1& \text{m2}\\
0& \text{otherwise}
\end{cases}$

- $x_{i,m3}=\begin{cases}
1& \text{m3}\\
0& \text{otherwise}
\end{cases}$
- $x_{i,m4}=\begin{cases}
1& \text{m4}\\
0& \text{otherwise}
\end{cases}$
- Possible in msqrob2 and MSstats but not possible with Perseus!



### Modeling and inference

#### RCB analysis
```{r warning=FALSE}
pe <- msqrob(
  object = pe,
  i = "protein",
  formula = ~ celltype + mouse)
```


#### CRD analysis 
```{r warning = FALSE}
pe2 <- msqrob(
  object = pe2,
  i = "protein",
  formula = ~ celltype)
```

#### Estimation, effect size and inference

Effect size in RCB
```{r}
library(ExploreModelMatrix)
VisualizeDesign(colData(pe),~ celltype + mouse)$plotlist
```

Effect size in CRD
```{r}
VisualizeDesign(colData(pe2),~ celltype)$plotlist
```


<details><summary> Click to see code for statistical inference </summary><p>
```{r}
L <- makeContrast("celltypeTreg = 0", parameterNames = c("celltypeTreg"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L)
pe2 <- hypothesisTest(object = pe2, i = "protein", contrast = L)
```
</p></details>

### Comparison of results

<details><summary> Click to see code </summary><p>
```{r warning=FALSE,echo=FALSE}
volcanoRCB <- ggplot(
    rowData(pe[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("RCB: \n", 
                sum(rowData(pe[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

volcanoCRD <- ggplot(
    rowData(pe2[["protein"]])$celltypeTreg,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("CRD: \n", 
                sum(rowData(pe2[["protein"]])$celltypeTreg$adjPval<0.05,na.rm=TRUE),
            " significant"))

xlims <- (
  range(
    rowData(pe[["protein"]])$celltypeTreg$logFC,
    rowData(pe2[["protein"]])$celltypeTreg$logFC,
              na.rm=TRUE) |> 
    abs() |> 
    max()
  ) * c(-1,1)

ylims <- range(-log10(rowData(pe[["protein"]])$celltypeTreg$pval), 
              -log10(rowData(pe2[["protein"]])$celltypeTreg$logFC), 
              na.rm=TRUE)
```
</p></details>
  
```{r warning=FALSE,echo=FALSE}
volcanoRCB + 
  xlim(xlims) + 
  ylim(ylims) 
volcanoCRD + 
  xlim(xlims) + 
  ylim(ylims)
```

### Comparison of standard deviation 

<details><summary> Click to see code </summary><p>
```{r}
accessions <- rownames(pe[["protein"]])[rownames(pe[["protein"]])%in%rownames(pe2[["protein"]])]
dat <- data.frame(
sigmaRBC = sapply(rowData(pe[["protein"]])$msqrobModels[accessions], getSigmaPosterior),
sigmaCRD <- sapply(rowData(pe2[["protein"]])$msqrobModels[accessions], getSigmaPosterior)
)

plotRBCvsCRD <- ggplot(data = dat, aes(sigmaRBC, sigmaCRD)) +
    geom_point(alpha = 0.1, shape = 20) +
    scale_x_log10() +
    scale_y_log10() +
    geom_abline(intercept=0,slope=1)
```
</p></details>

```{r}
  plotRBCvsCRD
```


- We clearly observe that the standard deviation of the protein expression in the RCB is smaller for the majority of the proteins than that obtained with the CRD

- Can you think of a reason why it would not be useful to block on a particular factor? 

## Pseudo-replication

- The Francisella tularensis that we used before was a subset of the data of Ramond et al. 
- The authors have run the sample for each bio-rep in technical triplicate on the mass-spectrometer.

### Data

<details><summary> Click to see code to import data </summary><p>

1. We use a peptides.txt file from MS-data quantified with maxquant that 
contains MS1 intensities summarized at the peptide level. 
```{r}
peptidesTable <- fread("https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/refs/heads/master/Francisella/data/maxquant/peptides.txt")
int64 <- which(sapply(peptidesTable,class) == "integer64")
for (j in int64) peptidesTable[[j]] <- as.numeric(peptidesTable[[j]])                      
```

2. Maxquant stores the intensity data for the different samples in columnns that start with Intensity. We can retreive the column names with the intensity data with the code below: 

```{r}
quantCols <- grep("Intensity ", names(peptidesTable))
```

4. Read the data and store it in  QFeatures object 

```{r}
pe <- readQFeatures(
  assayData = peptidesTable,
  fnames = 1,
  quantCols =  quantCols,
  name = "peptideRaw")
rm(peptidesTable)
gc()
gc()
```

5. Update data with information on design

```{r}
colData(pe)$genotype <- pe[[1]] |> 
  colnames() |> 
  substr(12,13) |>
  as.factor() |> 
  relevel("WT")

colData(pe)$biorep <- pe[[1]] |> 
  colnames() |> 
  substr(22,22)

colData(pe)$biorep <- paste(colData(pe)$genotype,colData(pe)$biorep,sep="_")
  
pe |> colData()
```

</p></details>

### Preprocessing

<details><summary> Click to see code to log-transfrom the data </summary><p>

1. Log transform

  - Peptides with zero intensities are missing peptides and should be represent
with a `NA` value rather than `0`.

```{r}
pe <- zeroIsNA(pe, "peptideRaw") # convert 0 to NA
```

  - Logtransform data with base 2

```{r}
pe <- logTransform(pe, base = 2, i = "peptideRaw", name = "peptideLog")
```

2. Filtering

  - We remove PSMs that could not be mapped to a protein or that map to multiple proteins (the protein identifier contains multiple identifiers separated by a `;`).

```{r}
pe <- filterFeatures(
    pe, ~ Proteins != "" & ## Remove failed protein inference
        !grepl(";", Proteins)) ## Remove protein groups
```

  - Remove reverse sequences (decoys) and contaminants. Note that this is indicated by the column names Reverse and depending on the version of maxQuant with Potential.contaminants or Contaminants.


```{r}
pe <- filterFeatures(pe,~Reverse != "+")
pe <- filterFeatures(pe,~ Contaminant != "+")
```

  - Drop peptides that were identified in less than three sample. We tolerate the following proportion of NAs: pNA = (n-3)/n. 

```{r}
nObs <- 3
n <- ncol(pe[["peptideLog"]])
pNA <- (n-nObs)/n
pe <- filterNA(pe, pNA = pNA, i = "peptideLog")
nrow(pe[["peptideLog"]])
```

We keep `r nrow(pe[["peptideLog"]])` peptides upon filtering.

3. Normalization by median centering

```{r}
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median")
```


4. Summarization. We use the standard sumarisation in aggregateFeatures, which is a
robust summarisation method.

```{r,warning=FALSE}
pe <- aggregateFeatures(pe,
    i = "peptideNorm", 
    fcol = "Proteins", 
    na.rm = TRUE,
    name = "protein")
```

</p></details>

```{r}
msqrob2gui:::plotPCA(pe, "protein", "biorep")
```


- Response?
- Experimental unit?
- Observational unit?
- Factors?

```{r}
VisualizeDesign(colData(pe),~ genotype + biorep)$plotlist
```

$\rightarrow$ Pseudo-replication, randomisation to bio-repeat and each bio-repeat measured in technical triplicate.
$\rightarrow$ If we would analyse the data using a linear model based on each measured intensity, we would act as if we had sampled 18 bio-repeats.
$\rightarrow$ Effect of interest has to be assessed between bio-repeats. So block analysis is not possible (if we condition on bio-repeat, we condition on the genotype)!


### Wrong analysis


```{r warning=FALSE}
pe <- msqrob(object = pe, i = "protein", formula = ~genotype, modelColumnName = "wrong", overwrite = TRUE)
L <- makeContrast("genotypeD8 = 0", parameterNames = c("genotypeD8"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L, modelColumn = "wrong",   resultsColumnNamePrefix = "wrong_", overwrite = TRUE)

volcanoWrong <- ggplot(
    rowData(pe[["protein"]])$wrong_genotypeD8,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("wrong analysis: ",sum(rowData(pe[["protein"]])$wrong_genotypeD8$adjPval <0.05, na.rm=TRUE)," DA"))

volcanoWrong
```


Note, that the analysis where we ignore that we have multiple technical repeats for each bio-repeat returns many significant DA proteins because we act as if we have much more independent observations.

### Correct analysis

- Mixed Models can model the correlation structure in the data

- They can acknowledge that protein expression values from runs from the same biological repeat are more alike than protein expression values from runs of different biological repeats. 

- Mixed models can also be used when inference between and within blocks is needed. 

- For the francisella example the formula in msqrob2 becomes: (formula: ~ genotype + (1|biorep)). 

$$ \left\{\begin{array}{rcl}
y_{ir} &=& \beta_0 + \beta_{wt}X_{wt,i} + b_i + \epsilon_{ir}\\
b_i & \sim & N(0,\sigma_b^2)\\
\epsilon_{ir} &\sim& N(0,\sigma_\epsilon^2)
\end{array}\right.$$



```{r}
pe <- msqrob(object = pe, i = "protein", formula = ~genotype + (1|biorep), overwrite = TRUE)
L <- makeContrast("genotypeD8 = 0", parameterNames = c("genotypeD8"))
pe <- hypothesisTest(object = pe, i = "protein", contrast = L, overwrite = TRUE)

volcanoMixed <- ggplot(
    rowData(pe[["protein"]])$genotypeD8,
    aes(x = logFC, y = -log10(pval), color = adjPval < 0.05)
) +
    geom_point(cex = 2.5) +
    scale_color_manual(values = alpha(c("black", "red"), 0.5)) +
    theme_minimal() +
    ggtitle(paste0("correct analysis: ",sum(rowData(pe[["protein"]])$genotypeD8$adjPval <0.05, na.rm=TRUE)," DA"))
volcanoMixed
```

- Note, that the statistical inference with mixed models is still a bit too liberal because it: 
    - hypothesis tests are only valid asymptotically (large number of biorepeats).
    - in small samples the between biorepeat variance is sometimes estimated to be zero due to estimation uncertainty and for these proteins the model still acts as if all 18 protein expression values are independent. 
    
- Mixed models are also very useful to analyse data from large labeled experiments [@vandenbulcke2025]. 

- But, they are beyond the scope of the lecture series. 

CONSULT BIOSTATISTICIAN IN CASE OF EXPERIMENTS WITH PSEUDO-REPLICATES, TECHNICAL REPEATS, COMPLEX DESIGNS, ...

# Software & code

- Our R/Bioconductor package [msqrob2](https://www.bioconductor.org/packages/release/bioc/html/msqrob2.html) can be used in R markdown scripts or with GUI/shinyApps [QFeaturesGUI](https://github.com/statOmics/QFeaturesGUI) (+) and [msqrob2gui](https://github.com/statOmics/msqrob2gui) (+: forked from the UCLouvain-CBIO lab).

- GUIs are intended as a introduction to the key concepts of proteomics data analysis for users who have no experience in R. 

- However, learning how to code data analyses in R markdown scripts is key for open en reproducible science and for reporting your proteomics data analyses and interpretation in a reproducible way. 


- More information on our tools can be found in our papers [@goeminne2016], [@goeminne2020], [@sticker2020] and [@vandenbulcke2025]. Please refer to our work when using our tools. 

<!--
- Clips on the code on importing the data and preprocessing can be found in [Part I Preprocessing](./pda_quantification_preprocessing.html)

- A clip on the code for modelling and statistical inference with msqrob2 is included below

<iframe width="560" height="315"
src="https://www.youtube.com/embed/eXxIdzGOPgY"
frameborder="0"
style="display: block; margin: auto;"
allow="autoplay; encrypted-media" allowfullscreen></iframe>
--> 

# References
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("pda_quantification_inference.Rmd");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
